---
title: "deseq2_re_analyses"
author: "Anshul Sinha"
date: "2/12/2022"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

```{r}
library(DESeq2)
library(dplyr)
library(tidyverse)
library(EnhancedVolcano)
```

```{r}

#read in the counts matrix 

raw_counts1_deseq2 <- read.csv("/Users/anshul.sinha/Desktop/Sequencing_analyses/phage_metagenomics_rainin_fall_2018/re-analyses_manuscript_data/deseq2/raw_counts_matrices/matrix_whole_expt_matrix.csv", header = TRUE)

raw_counts1_deseq2

#switching the columns and rows so that each row is a different contig and each column is a different sample 
switch_matrix_deseq2 <-  t(raw_counts1_deseq2)



switch_matrix_deseq2

write.csv(switch_matrix_deseq2, "/Users/anshul.sinha/Desktop/Sequencing_analyses/phage_metagenomics_rainin_fall_2018/re-analyses_manuscript_data/deseq2/transposed_matrices/transposed_matrix.csv") 




```




```{r}

#reading in manually edited switched matrix

raw_counts2_deseq2 <- read.csv("/Users/anshul.sinha/Desktop/Sequencing_analyses/phage_metagenomics_rainin_fall_2018/re-analyses_manuscript_data/deseq2/transposed_matrices/formatted_raw_counts.csv", check.names = FALSE)
raw_counts2_deseq2



#Formatting the rawcount2_deseq2 transposed matrix 



colnames(raw_counts2_deseq2)[1] <- "contig_ID"

raw_counts2_deseq2



#naming that column "contig_ID"

contig_ID_deseq2 <-  raw_counts2_deseq2$contig_ID

#Now I am generating a "sample index" so I can exclude "contig_ID" from our matrix 

sampleindex_deseq2 <- grepl("\\d", colnames(raw_counts2_deseq2))
##of the column names, anything that contains a numeric 


#converting to a matrix and only including the sample index columns 
raw_counts_matrix_deseq2 <- as.matrix(raw_counts2_deseq2[,sampleindex_deseq2])
raw_counts_matrix_deseq2


rownames(raw_counts_matrix_deseq2) <- contig_ID_deseq2

head(raw_counts_matrix_deseq2)

write.csv(raw_counts_matrix_deseq2, "/Users/anshul.sinha/Desktop/Sequencing_analyses/phage_metagenomics_rainin_fall_2018/re-analyses_manuscript_data/deseq2/raw_counts_formatted_transposed_matrices/matrix_for_input.csv")



```



```{r}
#reading in sample mapping file (coldata)
sample_mapping_deseq2 <- read.csv("/Users/anshul.sinha/Desktop/Sequencing_analyses/phage_metagenomics_rainin_fall_2018/re-analyses_manuscript_data/deseq2/sample_mapping/sample_mapping_deseq.csv", row.names = 1)

sample_mapping_deseq2

#changing all columns to factor 
sample_mapping_deseq2 <- mutate_if(sample_mapping_deseq2, is.character, as.factor)
sample_mapping_deseq2 <- mutate_if(sample_mapping_deseq2, is.integer, as.factor)
sample_mapping_deseq2

```


```{r}


#note that the row.names dont have a column name - this is consistent with how coldata is shown in the deseq2 vignette 


sample_mapping_deseq2


#Put the columns of the count data in the same order as rows names of the sample mapping, then make sure it worked (TRUE)

raw_counts_matrix_deseq2 <- raw_counts_matrix_deseq2[, unique(rownames(sample_mapping_deseq2))]


all(colnames(raw_counts_matrix_deseq2) == rownames(sample_mapping_deseq2))

sample_mapping_deseq2


#True if alignment is correct
```








```{r}
#starting the object construction with a simple design per vignette 

deseq2Data <- DESeqDataSetFromMatrix(countData=raw_counts_matrix_deseq2, colData=sample_mapping_deseq2, design = ~ Cage_n + Condition)


#since every gene contains at least one zero, we need to use estimatesizeFactors (type = poscounts)
#we can change our design later, since sizefactor estimation does not depend on design : https://www.biostars.org/p/271371/#:~:text=Dispersion%20estimation%20does%20depend%20on,but%20size%20factors%20do%20not.

deseq2Data <- estimateSizeFactors(deseq2Data, type= "poscounts")




```


```{r}
#relevelling the condition and treatment variables so that the ref is pre-gav - not sure this step is necessary because we are using contrast, but adding anyways-

#correction : relevelling the sample_mapping seems necessary, so that when we build the model.matrix, it contains the Treatment:Post_gav values and not Treatment:Pre_gav values 

#note that relevlling the deseq2Data or not DOES NOT make a difference ( I ran with and without )




sample_mapping_deseq2$Condition  <-relevel(sample_mapping_deseq2$Condition, ref="pre_gav")

levels(sample_mapping_deseq2$Condition)


deseq2Data$Condition  <-relevel(deseq2Data$Condition, ref="pre_gav")

levels(deseq2Data$Condition)


#creating more complex design - controlling for cage, looking at 


mm <- model.matrix(~ Treatment + Treatment:Cage_n + Treatment:Condition, sample_mapping_deseq2)
mm

```
















```{r}
#Running DESeq with the new model matrix 

deseq2Data = DESeq(deseq2Data, full = mm, betaPrior = FALSE)


```












```{r}

#removing the rows that did not converge in beta 

dds_clean <- deseq2Data[which(mcols(deseq2Data)$betaConv),]

```



```{r}
#contrasting to just show UCP vs. PBS with cutoffs 
UCP_vs_PBS_dds_results <- results(dds_clean, contrast=list("TreatmentUCP.Conditionpost_gav","TreatmentPBS.Conditionpost_gav"), lfcThreshold = 1)

UCP_vs_PBS_dds_results <- subset(UCP_vs_PBS_dds_results, padj < 0.01)

write.csv(UCP_vs_PBS_dds_results, "/Users/anshul.sinha/Desktop/Sequencing_analyses/phage_metagenomics_rainin_fall_2018/re-analyses_manuscript_data/deseq2/raw_outputs/UCP_v_PBS_0.01.csv")

#same as above, but not including cutoff so I can use this for the vol plot 

UCP_vs_PBS_dds_results1 <- results(dds_clean, contrast=list("TreatmentUCP.Conditionpost_gav","TreatmentPBS.Conditionpost_gav"))



```









```{r}
#contrasting to just show UCP vs. PBS with cutoffs  - same but with more strict cutoff
#UCP_vs_PBS_dds_results <- results(dds_clean, contrast=list("TreatmentUCP.Conditionpost_gav","TreatmentPBS.Conditionpost_gav"), lfcThreshold = 1)


#UCP_vs_PBS_dds_results <- subset(UCP_vs_PBS_dds_results, padj < 0.001)

#write.csv(UCP_vs_PBS_dds_results, "/Users/anshul.sinha/Desktop/Sequencing_analyses/phage_metagenomics_rainin_fall_2018/re-analyses_manuscript_data/deseq2/raw_outputs/UCP_v_PBS_0.001.csv")
```





```{r}
#contrasting to just show HP vs. PBS with cutoffs
HP_vs_PBS_dds_results <- results(dds_clean, contrast=list("TreatmentHP.Conditionpost_gav","TreatmentPBS.Conditionpost_gav"), lfcThreshold = 1)


HP_vs_PBS_dds_results <- subset(HP_vs_PBS_dds_results, padj < 0.01)

write.csv(HP_vs_PBS_dds_results, "/Users/anshul.sinha/Desktop/Sequencing_analyses/phage_metagenomics_rainin_fall_2018/re-analyses_manuscript_data/deseq2/raw_outputs/HP_v_PBS_0.01.csv")


#same as above, but not including cutoff so I can use this for the vol plot 
HP_vs_PBS_dds_results1 <- results(dds_clean, contrast=list("TreatmentHP.Conditionpost_gav","TreatmentPBS.Conditionpost_gav"))


```


```{r}
#contrasting to just show HP vs. PBS with cutoffs
#HP_vs_PBS_dds_results <- results(dds_clean, contrast=list("TreatmentHP.Conditionpost_gav","TreatmentPBS.Conditionpost_gav"), lfcThreshold = 1)


#HP_vs_PBS_dds_results <- subset(HP_vs_PBS_dds_results, padj < 0.001)

#write.csv(HP_vs_PBS_dds_results, "/Users/anshul.sinha/Desktop/Sequencing_analyses/phage_metagenomics_rainin_fall_2018/re-analyses_manuscript_data/deseq2/raw_outputs/HP_v_PBS_0.001.csv")



```




```{r}
#contrasting to just show UCP vs. HP with cutoffs
UCP_vs_HP_dds_results <- results(dds_clean, contrast=list("TreatmentUCP.Conditionpost_gav","TreatmentHP.Conditionpost_gav"), lfcThreshold = 1)

UCP_vs_HP_dds_results <- subset(UCP_vs_HP_dds_results, padj < 0.01)

write.csv(UCP_vs_HP_dds_results, "/Users/anshul.sinha/Desktop/Sequencing_analyses/phage_metagenomics_rainin_fall_2018/re-analyses_manuscript_data/deseq2/raw_outputs/UCP_v_HP_0.01.csv")


#same as above, but not including cutoff so I can use this for the vol plot 

UCP_vs_HP_dds_results1 <- results(dds_clean, contrast=list("TreatmentUCP.Conditionpost_gav","TreatmentHP.Conditionpost_gav"))

```


```{r}
#contrasting to just show UCP vs. HP with cutoffs
#UCP_vs_HP_dds_results <- results(dds_clean, contrast=list("TreatmentUCP.Conditionpost_gav","TreatmentHP.Conditionpost_gav"), lfcThreshold = 1)

#UCP_vs_HP_dds_results <- subset(UCP_vs_HP_dds_results, padj < 0.001)

#write.csv(UCP_vs_HP_dds_results, "/Users/anshul.sinha/Desktop/Sequencing_analyses/phage_metagenomics_rainin_fall_2018/re-analyses_manuscript_data/deseq2/raw_outputs/UCP_v_HP_0.001.csv")
```




```{r}
#reading in the contig metadata, so I can associate it with the deseq2 results 
contig_metadata_deseq2 <- read.csv("/Users/anshul.sinha/Desktop/Sequencing_analyses/phage_metagenomics_rainin_fall_2018/re-analyses_manuscript_data/metadata/0.7_cutoff/metadata.csv")

contig_metadata_deseq2
```



```{r}
#UCP  vs. PBS comparison - 0.01 cutoff 




UCP_vs_PBS_dds_results_df <- as.data.frame(UCP_vs_PBS_dds_results)
UCP_vs_PBS_dds_results_df <- UCP_vs_PBS_dds_results_df %>% 
  rownames_to_column(var = "ContigName")
#converting results table and naming first column so results table can be annotated



UCP_vs_PBS_dds_results_df_ordered <- UCP_vs_PBS_dds_results_df[order(UCP_vs_PBS_dds_results_df$pvalue),]




UCP_vs_PBS_dds_results_df_ordered_annotated <- left_join(UCP_vs_PBS_dds_results_df_ordered, contig_metadata_deseq2, by=c("ContigName"="ContigName"))


UCP_vs_PBS_dds_results_df_ordered_annotated_upreg <- subset(UCP_vs_PBS_dds_results_df_ordered_annotated,log2FoldChange > 0)
table(UCP_vs_PBS_dds_results_df_ordered_annotated_upreg$Family)



write_csv(UCP_vs_PBS_dds_results_df_ordered_annotated, "/Users/anshul.sinha/Desktop/Sequencing_analyses/phage_metagenomics_rainin_fall_2018/re-analyses_manuscript_data/deseq2/annotated_outputs/annotated_ucp_pbs.csv")

#now filtering so only those with crispr matches are shown 

UCP_vs_PBS_dds_results_df_ordered_annotated_crispr <-  UCP_vs_PBS_dds_results_df_ordered_annotated[!grepl("unknown", UCP_vs_PBS_dds_results_df_ordered_annotated$CRISPRHost),]


UCP_vs_PBS_dds_results_df_ordered_annotated_crispr


#now only get the ones that are upregulated 


UCP_vs_PBS_dds_results_df_ordered_annotated_crispr_upreg <- subset(UCP_vs_PBS_dds_results_df_ordered_annotated_crispr, log2FoldChange > 0) 

UCP_vs_PBS_dds_results_df_ordered_annotated_crispr_upreg

UCP_vs_PBS_dds_results_df_ordered_annotated_crispr_upreg_tab <- as.data.frame(table(UCP_vs_PBS_dds_results_df_ordered_annotated_crispr_upreg$CRISPRHost))
UCP_vs_PBS_dds_results_df_ordered_annotated_crispr_upreg_tab




write_csv(UCP_vs_PBS_dds_results_df_ordered_annotated_crispr, "/Users/anshul.sinha/Desktop/Sequencing_analyses/phage_metagenomics_rainin_fall_2018/re-analyses_manuscript_data/deseq2/annotated_outputs/crispr/ucp_pbs_crispr.csv")


write_csv(UCP_vs_PBS_dds_results_df_ordered_annotated_crispr_upreg_tab, "/Users/anshul.sinha/Desktop/Sequencing_analyses/phage_metagenomics_rainin_fall_2018/re-analyses_manuscript_data/deseq2/annotated_outputs/crispr/ucp_pbs_crispr_table.csv")


```



```{r}
#HP  vs. PBS comparison - 0.01 cutoff 




HP_vs_PBS_dds_results_df <- as.data.frame(HP_vs_PBS_dds_results)
HP_vs_PBS_dds_results_df <- HP_vs_PBS_dds_results_df %>% 
  rownames_to_column(var = "ContigName")
#converting results table and naming first column so results table can be annotated



HP_vs_PBS_dds_results_df_ordered <- HP_vs_PBS_dds_results_df[order(HP_vs_PBS_dds_results_df$pvalue),]




HP_vs_PBS_dds_results_df_ordered_annotated <- left_join(HP_vs_PBS_dds_results_df_ordered, contig_metadata_deseq2, by=c("ContigName"="ContigName"))



HP_vs_PBS_dds_results_df_ordered_annotated_upreg <- subset(HP_vs_PBS_dds_results_df_ordered_annotated,log2FoldChange > 0)
table(HP_vs_PBS_dds_results_df_ordered_annotated_upreg$Family)



write_csv(HP_vs_PBS_dds_results_df_ordered_annotated, "/Users/anshul.sinha/Desktop/Sequencing_analyses/phage_metagenomics_rainin_fall_2018/re-analyses_manuscript_data/deseq2/annotated_outputs/annotated_hp_pbs.csv")

HP_vs_PBS_dds_results_df_ordered_annotated_crispr <-  HP_vs_PBS_dds_results_df_ordered_annotated[!grepl("unknown", HP_vs_PBS_dds_results_df_ordered_annotated$CRISPRHost),]


HP_vs_PBS_dds_results_df_ordered_annotated_crispr_upreg <- subset(HP_vs_PBS_dds_results_df_ordered_annotated_crispr, log2FoldChange > 0) 

HP_vs_PBS_dds_results_df_ordered_annotated_crispr_upreg

HP_vs_PBS_dds_results_df_ordered_annotated_crispr_upreg_tab <- as.data.frame(table(HP_vs_PBS_dds_results_df_ordered_annotated_crispr_upreg$CRISPRHost))
HP_vs_PBS_dds_results_df_ordered_annotated_crispr_upreg_tab


HP_vs_PBS_dds_results_df_ordered_annotated_crispr

write_csv(HP_vs_PBS_dds_results_df_ordered_annotated_crispr, "/Users/anshul.sinha/Desktop/Sequencing_analyses/phage_metagenomics_rainin_fall_2018/re-analyses_manuscript_data/deseq2/annotated_outputs/crispr/hp_pbs_crispr.csv")

write_csv(HP_vs_PBS_dds_results_df_ordered_annotated_crispr_upreg_tab, "/Users/anshul.sinha/Desktop/Sequencing_analyses/phage_metagenomics_rainin_fall_2018/re-analyses_manuscript_data/deseq2/annotated_outputs/crispr/hp_pbs_crispr_tab.csv")




```



```{r}

#UCP  vs. HP comparison - 0.01 cutoff 

UCP_vs_HP_dds_results_df <- as.data.frame(UCP_vs_HP_dds_results)
UCP_vs_HP_dds_results_df <- UCP_vs_HP_dds_results_df %>% 
  rownames_to_column(var = "ContigName")
#converting results table and naming first column so results table can be annotated



UCP_vs_HP_dds_results_df_ordered <- UCP_vs_HP_dds_results_df[order(UCP_vs_HP_dds_results_df$pvalue),]




UCP_vs_HP_dds_results_df_ordered_annotated <- left_join(UCP_vs_HP_dds_results_df_ordered, contig_metadata_deseq2, by=c("ContigName"="ContigName"))





write_csv(UCP_vs_HP_dds_results_df_ordered_annotated, "/Users/anshul.sinha/Desktop/Sequencing_analyses/phage_metagenomics_rainin_fall_2018/re-analyses_manuscript_data/deseq2/annotated_outputs/annotated_ucp_hp.csv")





UCP_vs_HP_dds_results_df_ordered_annotated_crispr <-  UCP_vs_HP_dds_results_df_ordered_annotated[!grepl("unknown", UCP_vs_HP_dds_results_df_ordered_annotated$CRISPRHost),]

UCP_vs_HP_dds_results_df_ordered_annotated_crispr



write_csv(UCP_vs_HP_dds_results_df_ordered_annotated_crispr, "/Users/anshul.sinha/Desktop/Sequencing_analyses/phage_metagenomics_rainin_fall_2018/re-analyses_manuscript_data/deseq2/annotated_outputs/crispr/ucp_hp_crispr.csv")




```


```{r}
#now seeing how many of the differentially abundant upregulated scaffolds were found IN THE STOCK AND NOT BEFORE GAVAGE 

#read in the the df with scaffolds only found post gavage in UCP mice and also found in the UCP stock 

UCP_stock_mice_unique_post_gav <- read.csv("/Users/anshul.sinha/Desktop/Sequencing_analyses/phage_metagenomics_rainin_fall_2018/re-analyses_manuscript_data/stocks_analyses/UC_stocks/unique_scaf_post_gav_stock_UCP.csv")


#filtering for just upregulated dif abundant scaf 
UCP_vs_PBS_dds_results_df
UCP_vs_PBS_dds_results_df_upreg <- subset(UCP_vs_PBS_dds_results_df, log2FoldChange > 0)
UCP_vs_PBS_dds_results_df_upreg

intersect_deseq_uniquescaf_UCP <-  intersect(UCP_stock_mice_unique_post_gav$x, UCP_vs_PBS_dds_results_df_upreg$ContigName)
intersect_deseq_uniquescaf_UCP
length(intersect_deseq_uniquescaf_UCP)
```








```{r}
#now seeing how many of the differentially abundant upregulated scaffolds were found IN THE STOCK AND NOT BEFORE GAVAGE 

#read in the the df with scaffolds only found post gavage in UCP mice and also found in the UCP stock 

HP_stock_mice_unique_post_gav <- read.csv("/Users/anshul.sinha/Desktop/Sequencing_analyses/phage_metagenomics_rainin_fall_2018/re-analyses_manuscript_data/stocks_analyses/HP_stocks/unique_scaf_post_gav_stock_HP.csv")


#filtering for just upregulated dif abundant scaf 
HP_vs_PBS_dds_results_df
HP_vs_PBS_dds_results_df_upreg <- subset(HP_vs_PBS_dds_results_df, log2FoldChange > 0)
HP_vs_PBS_dds_results_df_upreg

intersect_deseq_uniquescaf_HP <-  intersect(HP_stock_mice_unique_post_gav$x, HP_vs_PBS_dds_results_df_upreg$ContigName)
intersect_deseq_uniquescaf_HP
length(intersect_deseq_uniquescaf_HP)
```


```{r}
#now... how many of these upregulated scaffolds are present in ALL 3 cages POST GAVAGE 

#1) READ IN A DF THAT CONTAINS THE ANNOTATED FEATURE TABLES 

sample_df <- read.csv("/Users/anshul.sinha/Desktop/Sequencing_analyses/phage_metagenomics_rainin_fall_2018/re-analyses_manuscript_data/coverage_converter/0.7/merged-geneious_output_0.7.csv")


#adding new column "treatment" based on VLP treatment 


sample_df <- sample_df %>% 
    mutate(Treatment = case_when(Sample == 'IDT_i5_10' ~ 'UCP_stock', Sample == 'IDT_i5_146' ~ 'UCP_stock', Sample == 'IDT_i5_158' ~ 'UCP_stock', Sample == 'IDT_i5_170' ~ 'UCP_stock', Sample == 'IDT_i5_57' ~ 'UCP_stock', Sample == 'IDT_i5_21' ~ 'HP_stock', Sample == 'IDT_i5_45' ~ 'HP_stock', Sample == 'IDT_i5_69' ~ 'HP_stock', Sample == 'IDT_i5_93' ~ 'HP_stock', Sample == 'IDT_i5_45' ~ 'HP_stock', str_sub(Sample, 1, 1) == 3 ~ 'UCP', str_sub(Sample, 1, 1) == 2 ~ 'HP', str_sub(Sample, 1, 1) == 1 ~ 'PBS', TRUE ~ 'NA'))
           

          
#adding new column "date" based on date

#first creating a function to take n characters from the end 

substrRight <- function(x, n){
  substr(x, nchar(x)-n+1, nchar(x))
}

#taking the last two characters fromt he sample collection and changing to the number of days from experiment start
#note that some of the stocks end with those 2 characters, first, create a character to label whether a sample comes from a stock

sample_df <- sample_df %>%
    mutate(Stock = case_when(substr(Sample, 1, 3) == 'IDT' ~ 'stock', TRUE ~ 'no'))


sample_df <- sample_df %>% 
    mutate(Date = case_when(substrRight(Sample, 2) == '03' ~ '7', substrRight(Sample, 2) == '10' ~ '14', substrRight(Sample, 2) == '17' ~ '21',substrRight(Sample, 2) == '18' ~ '22', substrRight(Sample, 2) == '22' ~ '26', substrRight(Sample, 2) == '25' ~ '29', substrRight(Sample, 2) == '02' ~ '37', substrRight(Sample, 2) == '04' ~ '39', Stock == 'yes' ~ 'NA', TRUE ~ 'NA'))


#now add 'Periods' based on these dates 

sample_df <- sample_df %>%
    mutate(Period = case_when(Date == '7'~ 'Colonization', Date == '14' ~ 'Colonization', Date == '21'~ 'Colonization', Date == '22'~ 'Gavage', Date == '26'~ 'Gavage', Date == '29'~ 'Gavage', Date == '37'~ 'DSS+washout', Date == '39'~ 'DSS+washout', TRUE ~ 'NA'))


#now add 'pre-VLP gaage' based on these dates 
sample_df <- sample_df %>%
    mutate(DSS = case_when(Date == '7'~ 'Pre-DSS', Date == '14' ~ 'Pre-DSS', Date == '21'~ 'Pre-DSS', Date == '22'~ 'Pre-DSS', Date == '26'~ 'Pre-DSS', Date == '29'~ 'Pre-DSS', Date == '37'~ 'Post-DSS', Date == '39'~ 'Post-DSS', TRUE ~ 'NA'))



#now add 'pre-VLP gaage' based on these dates 
sample_df <- sample_df %>%
    mutate(Gavage = case_when(Date == '7'~ 'Pre', Date == '14' ~ 'Pre', Date == '21'~ 'Pre', Date == '22'~ 'Post', Date == '26'~ 'Post', Date == '29'~ 'Post', Date == '37'~ 'Post', Date == '39'~ 'Post', TRUE ~ 'NA'))



#now add cage info 

sample_df <- sample_df %>% 
    mutate(Cage = case_when(substr(Sample, 1, 3) == '1-1' ~ 'cage1-1', substr(Sample, 1, 3) == '1-2' ~ 'cage1-2',substr(Sample, 1, 3) == '1-3' ~ 'cage1-3',substr(Sample, 1, 3) == '2-1' ~ 'cage2-1', substr(Sample, 1, 3) == '2-2' ~ 'cage2-2', substr(Sample, 1, 3) == '2-3' ~ 'cage2-3', substr(Sample, 1, 3) == '3-1' ~ 'cage3-1', substr(Sample, 1, 3) == '3-2' ~ 'cage3-2',  substr(Sample, 1, 3) == '3-3' ~ 'cage3-3', TRUE ~ 'NA'))



```



```{r}
#subset by cage 

cage21_df<- subset(sample_df, Cage == 'cage2-1')
cage22_df <- subset(sample_df, Cage == 'cage2-2')
cage23_df <- subset(sample_df, Cage == 'cage2-3')


cage31_df <- subset(sample_df, Cage == 'cage3-1')
cage32_df <- subset(sample_df, Cage == 'cage3-2')
cage33_df <- subset(sample_df, Cage == 'cage3-3')

#further subset to ensure we are dealing only with post gavage samples  

cage21_df_postg <- subset(cage21_df, Gavage == 'Post')
cage22_df_postg <- subset(cage22_df, Gavage == 'Post')
cage23_df_postg <- subset(cage23_df, Gavage == 'Post')


cage31_df_postg <- subset(cage31_df, Gavage == 'Post')
cage32_df_postg <- subset(cage32_df, Gavage == 'Post')
cage33_df_postg <- subset(cage33_df, Gavage == 'Post')

#further subset to get unique contigs for these 

cage21_df_postg_unique <- cage21_df_postg[!duplicated(cage21_df_postg$ContigName),]
cage22_df_postg_unique <- cage22_df_postg[!duplicated(cage22_df_postg$ContigName),]
cage23_df_postg_unique <- cage23_df_postg[!duplicated(cage23_df_postg$ContigName),]


cage31_df_postg_unique <- cage31_df_postg[!duplicated(cage31_df_postg$ContigName),]
cage32_df_postg_unique <- cage32_df_postg[!duplicated(cage32_df_postg$ContigName),]
cage33_df_postg_unique <- cage33_df_postg[!duplicated(cage33_df_postg$ContigName),]






```


```{r}
# 2) UCP vs. PBS  - ARE THE CONTIGS UPREGULATED FOUND IN ALL 3 CAGES ? 

#how many are in cage 1
upreg_cage1_UCP <- UCP_vs_PBS_dds_results_df_upreg$ContigName[UCP_vs_PBS_dds_results_df_upreg$ContigName %in% cage31_df_postg_unique$ContigName]

upreg_cage1_UCP <- as.data.frame(upreg_cage1_UCP)
colnames(upreg_cage1_UCP) <- c("ContigName")
upreg_cage1_UCP


#how many are in cage 1 and 2 

upreg_cage1_2_UCP <- upreg_cage1_UCP$ContigName[upreg_cage1_UCP$ContigName %in% cage32_df_postg_unique$ContigName]


upreg_cage1_2_UCP <- as.data.frame(upreg_cage1_2_UCP)
colnames(upreg_cage1_2_UCP) <- c("ContigName")
upreg_cage1_2_UCP


#how many are in cage 1 and 2 and 3 
upreg_cage1_2_3_UCP <- upreg_cage1_2_UCP$ContigName[upreg_cage1_2_UCP$ContigName %in% cage33_df_postg_unique$ContigName]

upreg_cage1_2_3_UCP <- as.data.frame(upreg_cage1_2_3_UCP)
colnames(upreg_cage1_2_3_UCP) <- c("ContigName")

upreg_cage1_2_3_UCP



```

```{r}
# 2) HP vs. PBS  - ARE THE CONTIGS UPREGULATED FOUND IN ALL 3 CAGES ? 

#how many are in cage 1
upreg_cage1_HP <- HP_vs_PBS_dds_results_df_upreg$ContigName[HP_vs_PBS_dds_results_df_upreg$ContigName %in% cage21_df_postg_unique$ContigName]

upreg_cage1_HP <- as.data.frame(upreg_cage1_HP)
colnames(upreg_cage1_HP) <- c("ContigName")
upreg_cage1_HP


#how many are in cage 1 and 2 

upreg_cage1_2_HP <- upreg_cage1_HP$ContigName[upreg_cage1_HP$ContigName %in% cage22_df_postg_unique$ContigName]


upreg_cage1_2_HP <- as.data.frame(upreg_cage1_2_HP)
colnames(upreg_cage1_2_HP) <- c("ContigName")
upreg_cage1_2_HP


#how many are in cage 1 and 2 and 3 
upreg_cage1_2_3_HP <- upreg_cage1_2_HP$ContigName[upreg_cage1_2_HP$ContigName %in% cage23_df_postg_unique$ContigName]

upreg_cage1_2_3_HP <- as.data.frame(upreg_cage1_2_3_HP)
colnames(upreg_cage1_2_3_HP) <- c("ContigName")



upreg_cage1_2_3_HP 





```

```{r}
#other stats 


#upregulated UCP scaf also upregulated in HP 
upreg_both <-  HP_vs_PBS_dds_results_df_upreg$ContigName[HP_vs_PBS_dds_results_df_upreg$ContigName %in% UCP_vs_PBS_dds_results_df_upreg$ContigName]


upreg_both <- as.data.frame(upreg_both)
colnames(upreg_both) <- c("ContigName")

upreg_both

#create 'notin' operator - https://www.r-bloggers.com/2018/07/the-notin-operator/
`%notin%` <- Negate(`%in%`)

#Only those upregulated in UCP scaf :

upreg_UCP_only <-  UCP_vs_PBS_dds_results_df_upreg$ContigName[UCP_vs_PBS_dds_results_df_upreg$ContigName %notin% HP_vs_PBS_dds_results_df_upreg$ContigName]
upreg_UCP_only <- as.data.frame(upreg_UCP_only)
colnames(upreg_UCP_only) <- c("ContigName")

upreg_UCP_only

#Only those upregulated in HP scaf :

upreg_HP_only <-  HP_vs_PBS_dds_results_df_upreg$ContigName[HP_vs_PBS_dds_results_df_upreg$ContigName %notin% UCP_vs_PBS_dds_results_df_upreg$ContigName]
upreg_HP_only <- as.data.frame(upreg_HP_only)
colnames(upreg_HP_only) <- c("ContigName")

upreg_HP_only

#how many of those are found in all 3 cages - UCP
length(intersect(upreg_UCP_only$ContigName, upreg_cage1_2_3_UCP$ContigName))


#how many of those are found in all 3 cages - HP
length(intersect(upreg_HP_only$ContigName, upreg_cage1_2_3_HP$ContigName))
```
```{r}

#volcano plot: UCvs.PBS

#in order to plot the cutoff of padj and not p-value, we need to give the pvalue column in the df the values from the padj column 


UCP_vs_PBS_dds_results1_changed <- UCP_vs_PBS_dds_results1
UCP_vs_PBS_dds_results1_changed$pvalue <- UCP_vs_PBS_dds_results1_changed$padj
#write.csv(UCP_vs_PBS_dds_results1_changed, "/Users/anshul.sinha/Desktop/Sequencing_analyses/phage_metagenomics_rainin_fall_2018/re-analyses_manuscript_data/deseq2/test_changed_df.csv")
#write.csv(UCP_vs_PBS_dds_results1, "/Users/anshul.sinha/Desktop/Sequencing_analyses/phage_metagenomics_rainin_fall_2018/re-analyses_manuscript_data/deseq2/test_df.csv")


keyvals_UCP_PBS <- ifelse(
    UCP_vs_PBS_dds_results1_changed$log2FoldChange < -1 & UCP_vs_PBS_dds_results1_changed$padj <0.01, 'black',
      ifelse(UCP_vs_PBS_dds_results1_changed$log2FoldChange > 1 & UCP_vs_PBS_dds_results1_changed$padj <0.01, 'red',
        'light grey'))
  keyvals_UCP_PBS[is.na(keyvals_UCP_PBS)] <- 'light grey'
  names(keyvals_UCP_PBS)[keyvals_UCP_PBS == 'red'] <- 'Over-Abundant UC VLP Scaffolds'
  names(keyvals_UCP_PBS)[keyvals_UCP_PBS == 'light grey'] <- 'Non-significant'
  names(keyvals_UCP_PBS)[keyvals_UCP_PBS == 'black'] <- 'Under-Abundant UC VLP Scaffolds'

EV_UCPBS <- EnhancedVolcano(UCP_vs_PBS_dds_results1_changed, lab=NA , x = 'log2FoldChange',
    y = 'pvalue', pCutoff = 0.01, selectLab = rownames(UCP_vs_PBS_dds_results1_changed)[which(names(keyvals_UCP_PBS) %in% c('Over-Abundant UC VLP Scaffolds', 'Under-Abundant UC VLP Scaffolds'))], pointSize = 2, gridlines.major = FALSE, gridlines.minor = FALSE, colCustom = keyvals_UCP_PBS,  legendPosition = 'right', legendLabSize = 12,legendIconSize = 4.0, title = "UCbac + UC VLPs vs. UCbac + PBS",     subtitle = "", caption = ""
)

EV_UCPBS


ggsave("/Users/anshul.sinha/Desktop/Sequencing_analyses/phage_metagenomics_rainin_fall_2018/re-analyses_manuscript_data/deseq2/vol_plots/uc_pbs.tiff", width = 7, height =8, dpi=300)

```


```{r}



#volcano plot:  HP vs.PBS

#in order to plot the cutoff of padj and not p-value, we need to give the pvalue column in the df the values from the padj column 


HP_vs_PBS_dds_results1_changed  <- HP_vs_PBS_dds_results1
HP_vs_PBS_dds_results1_changed$pvalue <- HP_vs_PBS_dds_results1_changed$padj

HP_vs_PBS_dds_results1_changed


keyvals_HP_PBS <- ifelse(
    HP_vs_PBS_dds_results1_changed$log2FoldChange < -1 & HP_vs_PBS_dds_results1_changed$padj <0.01, 'black',
      ifelse(HP_vs_PBS_dds_results1_changed$log2FoldChange > 1 & HP_vs_PBS_dds_results1_changed$padj <0.01, 'blue',
        'light grey'))
  keyvals_HP_PBS[is.na(keyvals_HP_PBS)] <- 'light grey'
  names(keyvals_HP_PBS)[keyvals_HP_PBS == 'blue'] <- 'Over-Abundant HP VLP Scaffolds'
  names(keyvals_HP_PBS)[keyvals_HP_PBS == 'light grey'] <- 'Non-significant'
  names(keyvals_HP_PBS)[keyvals_HP_PBS == 'black'] <- 'Under-Abundant HP VLP Scaffolds'

EV_gav_HP_PBS <- EnhancedVolcano(HP_vs_PBS_dds_results1_changed, lab=NA , x = 'log2FoldChange',
    y = 'pvalue', pCutoff = 0.01, selectLab = rownames(HP_vs_PBS_dds_results1_changed)[which(names(keyvals_HP_PBS) %in% c('Over-Abundant HP VLP Scaffolds', 'Under-Abundant HP VLP Scaffolds'))], pointSize = 2, gridlines.major = FALSE, gridlines.minor = FALSE, colCustom = keyvals_HP_PBS,  legendPosition = 'right', legendLabSize = 12,legendIconSize = 4.0, title = "UCbac + HP VLPs vs. UCbac + PBS",     subtitle = "", caption = ""
)

EV_gav_HP_PBS


ggsave("/Users/anshul.sinha/Desktop/Sequencing_analyses/phage_metagenomics_rainin_fall_2018/re-analyses_manuscript_data/deseq2/vol_plots/HP_PBS.tiff", width = 7, height =8, dpi=300)

```



