---
title: "mouse-diversity_stock_lib_manuscript"
output: html_document
editor_options: 
  chunk_output_type: inline
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

```{r}
install.packages("vegan")
```


```{r}
library(vegan)
library(RColorBrewer)
library(ggplot2)
```



```{r}
#NMDS comparison of VCs during bacterial colonization period 

my.data.colonization.vcs <- read.csv("/Users/anshul.sinha/Desktop/Sequencing_analyses/phage_metagenomics_rainin_fall_2018/manuscript_data/matrices/rel_abundance_matrix/vc/mouse/colonization/matrix_vc_col.csv", row.names = 1)


treatment=c(rep("PBS",7),rep("healthy", 6), rep("UC",8)) 
colors=c(rep("black",7),rep("blue",6), rep("red",8))


treatment.df.colonization.vc <-data.frame(treatment=c(rep("PBS",7),rep("healthy",6),rep("UC",8)), my.data.colonization.vcs)

treatment.df.colonization.vc

my.matrix.colonization.vcs <- as.matrix(my.data.colonization.vcs)

NMDS.colonization.vcs <- metaMDS(my.matrix.colonization.vcs, k=2)



col_vcsplot <-ordiplot(NMDS.colonization.vcs,type="n", cex.axis =2)

points(NMDS.colonization.vcs, col=colors, pch=16, cex = 4 )


col_vcsplot


```

```{r}
#Comparison of PBS control and UCP treatment during colonization (VC)

my.data.col.npuc.vc <- read.csv("/Users/anshul.sinha/Desktop/Sequencing_analyses/phage_metagenomics_rainin_fall_2018/manuscript_data/matrices/rel_abundance_matrix/vc/mouse/colonization/matrix_vc_np_uc.csv", row.names = 1)


treatment=c(rep("no phage",7),rep("UC", 8)) 
colors=c(rep("black",7),rep("red",8))


treatment.df.col.npuc.vc <-data.frame(treatment=c(rep("no phage",7),rep("UC",8)), my.data.col.npuc.vc)

treatment.df.col.npuc.vc

my.matrix.col.npuc.vc <- as.matrix(my.data.col.npuc.vc)

NMDS.col.npuc.vc <- metaMDS(my.matrix.col.npuc.vc, k=2)






BC.col.scaffolds.npuc.vc <-vegdist(my.matrix.col.npuc.vc, method="bray", binary=FALSE, diag=FALSE, upper=FALSE, na.rm = FALSE)
adonis2(formula = BC.col.scaffolds.npuc.vc ~ treatment, data = treatment.df.col.npuc.vc , permutations = 999)

BC.col.scaffolds.npuc.vc<- set.seed(BC.col.scaffolds.npuc.vc)

```





```{r}
#Comparison of PBS control and HP treatment during bacterial colonization period (VC)

my.data.col.nphp.vc <- read.csv("/Users/anshul.sinha/Desktop/Sequencing_analyses/phage_metagenomics_rainin_fall_2018/manuscript_data/matrices/rel_abundance_matrix/vc/mouse/colonization/matrix_vc_col_np_hp.csv", row.names = 1)


treatment=c(rep("no phage",7),rep("healthy", 6)) 
colors=c(rep("black",7),rep("blue",6))


treatment.df.col.nphp.vc <-data.frame(treatment=c(rep("no phage",7),rep("healthy",6)), my.data.col.nphp.vc)

treatment.df.col.nphp.vc

my.matrix.col.nphp.vc <- as.matrix(my.data.col.nphp.vc)

NMDS.col.nphp.vc <- metaMDS(my.matrix.col.nphp.vc, k=2)


ordiplot(NMDS.col.nphp.vc,type="n")

points(NMDS.col.nphp.vc, col=colors, pch=16 )



BC.col.scaffolds.nphp.vc <-vegdist(my.matrix.col.nphp.vc, method="bray", binary=FALSE, diag=FALSE, upper=FALSE, na.rm = FALSE)
adonis2(formula = BC.col.scaffolds.nphp.vc ~ treatment, data = treatment.df.col.nphp.vc , permutations = 999)

BC.col.scaffolds.nphp.vc <- set.seed(BC.col.scaffolds.nphp.vc)
#np vs hp colonization vc 
```

```{r}
#Comparison of HP and UCP treatments during colonization (VC)

my.data.col.hpuc.vc <- read.csv("/Users/anshul.sinha/Desktop/Sequencing_analyses/phage_metagenomics_rainin_fall_2018/manuscript_data/matrices/rel_abundance_matrix/vc/mouse/colonization/matrix_vc_col_hp_uc.csv", row.names = 1)




treatment=c(rep("healthy", 6),rep("UC phage", 8)) 
colors=c(rep("blue", 6),rep("red", 8))


treatment.df.col.hpuc.vc <-data.frame(treatment=c(rep("healthy",6),rep("UC phage",8)), my.data.col.hpuc.vc)

treatment.df.col.hpuc.vc

my.matrix.col.hpuc.vc <- as.matrix(my.data.col.hpuc.vc)

NMDS.col.hpuc.vc <- metaMDS(my.matrix.col.hpuc.vc, k=2)


ordiplot(NMDS.col.hpuc.vc,type="points")
orditorp(NMDS.col.hpuc.vc, display="sites")
#used orditorp to try to declutter NMDS

points(NMDS.col.hpuc.vc, col=colors, pch=16, cex =2, data=my.data.col.hpuc.vc)

text(NMDS.col.hpuc.vc, cex =0.6, pos=4)


BC.col.scaffolds.hpuc.vc <-vegdist(my.matrix.col.hpuc.vc, method="bray", binary=FALSE, diag=FALSE, upper=FALSE, na.rm = FALSE)
adonis2(formula = BC.col.scaffolds.hpuc.vc ~ treatment, data = treatment.df.col.hpuc.vc , permutations = 999)

BC.col.scaffolds.hpuc.vc <- set.seed(BC.col.scaffolds.hpuc.vc)

```

```{r}
BC.col.vcs <-vegdist(my.matrix.colonization.vcs, method="bray", binary=FALSE, diag=FALSE, upper=FALSE, na.rm = FALSE,)
adonis2(formula = BC.col.vcs ~ treatment, data = treatment.df.colonization.vc , permutations = 999)

BC.col.vcs <- set.seed(BC.col.vcs)

```

```{r}
#Comparison of phage gavage period VCs between treatment groups
my.data.gavage.vcs <- read.csv("/Users/anshul.sinha/Desktop/Sequencing_analyses/phage_metagenomics_rainin_fall_2018/manuscript_data/matrices/rel_abundance_matrix/vc/mouse/gavage/matrix_vc_gav.csv", row.names = 1)


treatment=c(rep("PBS",7),rep("healthy", 9), rep("UC",9)) 
colors=c(rep("black",7),rep("blue",9), rep("red",9))


treatment.df.gavage.vc <-data.frame(treatment=c(rep("PBS",7),rep("healthy",9),rep("UC",9)), my.data.gavage.vcs)

treatment.df.gavage.vc

my.matrix.gavage.vcs <- as.matrix(my.data.gavage.vcs)

NMDS.gavage.vcs <- metaMDS(my.matrix.gavage.vcs, k=2)


ordiplot(NMDS.gavage.vcs,type="n", cex.axis=2)

points(NMDS.gavage.vcs, col=colors, pch=16, cex =4 )
```

```{r}
BC.gav.vcs <-vegdist(my.matrix.gavage.vcs, method="bray", binary=FALSE, diag=FALSE, upper=FALSE, na.rm = FALSE,)
adonis2(formula = BC.gav.vcs ~ treatment, data = treatment.df.gavage.vc , permutations = 999)
BC.gav.vcs <- set.seed(BC.gav.vcs)
```

```{r}
#Comparing PBS control and UCP treatment during phage gavage period (VCs)


my.data.gav.npuc.vc <- read.csv("/Users/anshul.sinha/Desktop/Sequencing_analyses/phage_metagenomics_rainin_fall_2018/manuscript_data/matrices/rel_abundance_matrix/vc/mouse/gavage/matrix_vc_gav_np_uc.csv", row.names = 1)


treatment=c(rep("no phage",7),rep("UC", 9)) 
colors=c(rep("black",7),rep("red",9))


treatment.df.gav.npuc.vc <-data.frame(treatment=c(rep("no phage",7),rep("UC",9)), my.data.gav.npuc.vc)

treatment.df.gav.npuc.vc

my.matrix.gav.npuc.vc <- as.matrix(my.data.gav.npuc.vc)

NMDS.gav.npuc.vc <- metaMDS(my.matrix.gav.npuc.vc, k=2)



BC.gav.scaffolds.npuc.vc <-vegdist(my.matrix.gav.npuc.vc, method="bray", binary=FALSE, diag=FALSE, upper=FALSE, na.rm = FALSE)
adonis2(formula = BC.gav.scaffolds.npuc.vc ~ treatment, data = treatment.df.gav.npuc.vc , permutations = 999)

BC.gav.scaffolds.npuc.vc <- set.seed(BC.gav.scaffolds.npuc.vc)
#np vs uc gavage vc 
```

```{r}

#Comparing PBS control and HP treatments during phage gavage period (VCs)

my.data.gav.nphp.vc <- read.csv("/Users/anshul.sinha/Desktop/Sequencing_analyses/phage_metagenomics_rainin_fall_2018/manuscript_data/matrices/rel_abundance_matrix/vc/mouse/gavage/matrix_vc_gav_np_hp.csv", row.names = 1)


treatment=c(rep("no phage",7),rep("healthy", 9)) 
colors=c(rep("black",7),rep("blue",9))


treatment.df.gav.nphp.vc <-data.frame(treatment=c(rep("no phage",7),rep("healthy",9)), my.data.gav.nphp.vc)

treatment.df.gav.nphp.vc

my.matrix.gav.nphp.vc <- as.matrix(my.data.gav.nphp.vc)

NMDS.gav.nphp.vc <- metaMDS(my.matrix.gav.nphp.vc, k=2)


ordiplot(NMDS.gav.nphp.vc,type="n")

points(NMDS.gav.nphp.vc, col=colors, pch=16 )



BC.gav.scaffolds.nphp.vc <-vegdist(my.matrix.gav.nphp.vc, method="bray", binary=FALSE, diag=FALSE, upper=FALSE, na.rm = FALSE)
adonis2(formula = BC.gav.scaffolds.nphp.vc ~ treatment, data = treatment.df.gav.nphp.vc , permutations = 999)

BC.gav.scaffolds.nphp.vc <- set.seed(BC.gav.scaffolds.nphp.vc)
#np vs hp gavage vc 

```

```{r}
#comparing HP and UCP treatments during phage gavage period (VCs)

my.data.gav.hpuc.vc <- read.csv("/Users/anshul.sinha/Desktop/Sequencing_analyses/phage_metagenomics_rainin_fall_2018/manuscript_data/matrices/rel_abundance_matrix/vc/mouse/gavage/matrix_vc_gav_hp_uc.csv", row.names = 1)




treatment=c(rep("healthy", 9),rep("UC phage", 9)) 
colors=c(rep("blue", 9),rep("red", 9))


treatment.df.gav.hpuc.vc <-data.frame(treatment=c(rep("healthy",9),rep("UC phage",9)), my.data.gav.hpuc.vc)

treatment.df.gav.hpuc.vc

my.matrix.gav.hpuc.vc <- as.matrix(my.data.gav.hpuc.vc)

NMDS.gav.hpuc.vc <- metaMDS(my.matrix.gav.hpuc.vc, k=2)


ordiplot(NMDS.gav.hpuc.vc,type="n")


points(NMDS.gav.hpuc.vc, col=colors, pch=16)



BC.gav.scaffolds.hpuc.vc <-vegdist(my.matrix.gav.hpuc.vc, method="bray", binary=FALSE, diag=FALSE, upper=FALSE, na.rm = FALSE)
adonis2(formula = BC.gav.scaffolds.hpuc.vc ~ treatment, data = treatment.df.gav.hpuc.vc , permutations = 999)

BC.gav.scaffolds.hpuc.vc <- set.seed(BC.gav.scaffolds.hpuc.vc)
#hp vs uc gavage scaffolds 
```



```{r}
#Comparing VCs during DSS washout period between treatment groups


my.data.dss.vcs <- read.csv("/Users/anshul.sinha/Desktop/Sequencing_analyses/phage_metagenomics_rainin_fall_2018/manuscript_data/matrices/rel_abundance_matrix/vc/mouse/dss/matrix_dss.csv", row.names = 1)


treatment=c(rep("PBS",6),rep("healthy", 5), rep("UC",6)) 
colors=c(rep("black",6),rep("blue",5), rep("red",6))


treatment.df.dss.vc <-data.frame(treatment=c(rep("PBS",6),rep("healthy",5),rep("UC",6)), my.data.dss.vcs)

treatment.df.dss.vc

my.matrix.dss.vcs <- as.matrix(my.data.dss.vcs)

NMDS.dss.vcs <- metaMDS(my.matrix.dss.vcs, k=2)


ordiplot(NMDS.dss.vcs,type="n", cex.axis=2)

points(NMDS.dss.vcs, col=colors, pch=16, cex =4)
```



```{r}
#Comparing PBS control and UCP treatment during DSS washout period (VCs)
my.data.dss.npuc.vc <- read.csv("/Users/anshul.sinha/Desktop/Sequencing_analyses/phage_metagenomics_rainin_fall_2018/manuscript_data/matrices/rel_abundance_matrix/vc/mouse/dss/matrix_dss_np_uc.csv", row.names = 1)


treatment=c(rep("no phage",6),rep("UC", 6)) 
colors=c(rep("black",6),rep("red",6))


treatment.df.dss.npuc.vc <-data.frame(treatment=c(rep("no phage",6),rep("UC",6)), my.data.dss.npuc.vc)

treatment.df.dss.npuc.vc

my.matrix.dss.npuc.vc <- as.matrix(my.data.dss.npuc.vc)

NMDS.dss.npuc.vc <- metaMDS(my.matrix.dss.npuc.vc, k=2)



BC.dss.scaffolds.npuc.vc <-vegdist(my.matrix.dss.npuc.vc, method="bray", binary=FALSE, diag=FALSE, upper=FALSE, na.rm = FALSE)
adonis2(formula = BC.dss.scaffolds.npuc.vc ~ treatment, data = treatment.df.dss.npuc.vc , permutations = 999)

BC.dss.scaffolds.npuc.vc <- set.seed(BC.dss.scaffolds.npuc.vc)
#np vs uc gavage scaffolds 
```


```{r}
#Comparison of PBS control and HP treatment during DSS washout (VCs)


my.data.dss.nphpvcs <- read.csv("/Users/anshul.sinha/Desktop/Sequencing_analyses/phage_metagenomics_rainin_fall_2018/manuscript_data/matrices/rel_abundance_matrix/vc/mouse/dss/matrix_dss_np_hp.csv", row.names = 1)


treatmentnphp=c(rep("PBS",6),rep("healthy", 5)) 
colorsnphp=c(rep("black",6),rep("blue",5))


treatment.df.dssnphp.vc <-data.frame(treatment=c(rep("PBS",6),rep("healthy",5)), my.data.dss.nphpvcs)

treatment.df.dssnphp.vc

my.matrix.dss.vcsnphp <- as.matrix(my.data.dss.nphpvcs)

NMDS.dss.vcsnphp <- metaMDS(my.matrix.dss.vcsnphp, k=2)


ordiplot(NMDS.dss.vcsnphp,type="n")

points(NMDS.dss.vcsnphp, col=colors, pch=16 )
```

```{r}
BC.dss.nphpvcs <-vegdist(my.matrix.dss.vcsnphp, method="bray", binary=FALSE, diag=FALSE, upper=FALSE, na.rm = FALSE,)
adonis2(formula = BC.dss.nphpvcs ~ treatment, data = treatment.df.dssnphp.vc , permutations = 999)

BC.dss.nphpvcs <- set.seed(BC.dss.nphpvcs)

```



```{r}
#Comparing UCP and HP treatments during DSS washout period (VCs)

my.data.dss.uchpvcs <- read.csv("/Users/anshul.sinha/Desktop/Sequencing_analyses/phage_metagenomics_rainin_fall_2018/manuscript_data/matrices/rel_abundance_matrix/vc/mouse/dss/matrix_dss_hp_uc.csv", row.names = 1)


treatment=c(rep("healthy",5),rep("UC", 6)) 
colors=c(rep("blue",5),rep("red",6))


treatment.df.dssuchp.vc <-data.frame(treatment=c(rep("healthy",5),rep("UC",6)), my.data.dss.uchpvcs)

treatment.df.dssuchp.vc

my.matrix.dss.vcsuchp <- as.matrix(my.data.dss.uchpvcs)

NMDS.dss.vcsuchp <- metaMDS(my.matrix.dss.vcsuchp, k=2)


ordiplot(NMDS.dss.vcsuchp,type="n")

points(NMDS.dss.vcsuchp, col=colors, pch=16 )
```

```{r}
BC.dss.uchpvcs <-vegdist(my.matrix.dss.vcsuchp, method="bray", binary=FALSE, diag=FALSE, upper=FALSE, na.rm = FALSE,)
adonis2(formula = BC.dss.uchpvcs ~ treatment, data = treatment.df.dssuchp.vc , permutations = 999)
BC.dss.uchpvcs <- set.seed(BC.dss.uchpvcs)
```





```{r}

#Comparing scaffold differences during bacterial colonization period between treatment groups

my.data.colonization.scaffolds <- read.csv("/Users/anshul.sinha/Desktop/Sequencing_analyses/phage_metagenomics_rainin_fall_2018/manuscript_data/matrices/rel_abundance_matrix/scaffolds/colonization/matrix_scaffolds_col.csv", row.names = 1)


treatment=c(rep("PBS",7),rep("healthy", 6), rep("UC",8)) 
colors=c(rep("black",7),rep("blue",6), rep("red",8))


treatment.df.colonization.scaffolds <-data.frame(treatment=c(rep("PBS",7),rep("healthy",6),rep("UC",8)), my.data.colonization.scaffolds)

treatment.df.colonization.scaffolds

my.matrix.colonization.scaffolds <- as.matrix(my.data.colonization.scaffolds)

NMDS.colonization.scaffolds <- metaMDS(my.matrix.colonization.scaffolds, k=2)


ordiplot(NMDS.colonization.scaffolds,type="n", cex.axis= 2)

points(NMDS.colonization.scaffolds, col=colors, pch=16, cex=4)

#ordiellipse(NMDS.gavage.scaffolds,groups=treatment,col=c(rep("black",7),rep("blue",9),rep("red",9)),draw="polygon",alpha=0.4,border=c(rep("black",7),rep("blue",9),rep("red",9)), conf = 0.95)

```



```{r}
BC.col.scaffolds <-vegdist(my.matrix.colonization.scaffolds, method="bray", binary=FALSE, diag=FALSE, upper=FALSE, na.rm = FALSE,)
adonis2(formula = BC.col.scaffolds ~ treatment, data = treatment.df.colonization.scaffolds , permutations = 999)

BC.col.scaffolds <- set.seed(BC.col.scaffolds)
```


```{r}
#Comparing PBS control and UCP treatment during colonization period (scaffolds)
my.data.col.npuc.scaf <- read.csv("/Users/anshul.sinha/Desktop/Sequencing_analyses/phage_metagenomics_rainin_fall_2018/manuscript_data/matrices/rel_abundance_matrix/scaffolds/colonization/matrix_scaffolds_col_np_uc.csv", row.names = 1)


treatment=c(rep("no phage",7),rep("UC", 8)) 
colors=c(rep("black",7),rep("red",8))


treatment.df.col.npuc.scaf <-data.frame(treatment=c(rep("no phage",7),rep("UC",8)), my.data.col.npuc.scaf)

treatment.df.col.npuc.scaf

my.matrix.col.npuc.scaf <- as.matrix(my.data.col.npuc.scaf)

NMDS.col.npuc.scaf <- metaMDS(my.matrix.col.npuc.scaf, k=2)






BC.col.scaffolds.npuc.scaf <-vegdist(my.matrix.col.npuc.scaf, method="bray", binary=FALSE, diag=FALSE, upper=FALSE, na.rm = FALSE)
adonis2(formula = BC.col.scaffolds.npuc.scaf ~ treatment, data = treatment.df.col.npuc.scaf , permutations = 999)

BC.col.scaffolds.npuc.scaf <- set.seed(BC.col.scaffolds.npuc.scaf)
#np vs uc colonization scaffolds 
```


```{r}

#Comparing PBS control and HP treatment during bacterial colonization period (scaffolds)

my.data.col.nphp.scaf <- read.csv("/Users/anshul.sinha/Desktop/Sequencing_analyses/phage_metagenomics_rainin_fall_2018/manuscript_data/matrices/rel_abundance_matrix/scaffolds/colonization/matrix_scaffolds_np_hp.csv", row.names = 1)


treatment=c(rep("no phage",7),rep("healthy", 6)) 
colors=c(rep("black",7),rep("blue",6))


treatment.df.col.nphp.scaf <-data.frame(treatment=c(rep("no phage",7),rep("healthy",6)), my.data.col.nphp.scaf)

treatment.df.col.nphp.scaf

my.matrix.col.nphp.scaf <- as.matrix(my.data.col.nphp.scaf)

NMDS.col.nphp.scaf <- metaMDS(my.matrix.col.nphp.scaf, k=2)


ordiplot(NMDS.col.nphp.scaf,type="n")

points(NMDS.col.nphp.scaf, col=colors, pch=16 )



BC.col.scaffolds.nphp.scaf <-vegdist(my.matrix.col.nphp.scaf, method="bray", binary=FALSE, diag=FALSE, upper=FALSE, na.rm = FALSE)
adonis2(formula = BC.col.scaffolds.nphp.scaf ~ treatment, data = treatment.df.col.nphp.scaf , permutations = 999)

BC.col.scaffolds.nphp.scaf <- set.seed(BC.col.scaffolds.nphp.scaf)
#np vs hp colonization scaffolds 
```


```{r}
#Comparison of HP and UCP treatment during bacterial colonization period (scaffolds)

my.data.col.hpuc.scaf <- read.csv("/Users/anshul.sinha/Desktop/Sequencing_analyses/phage_metagenomics_rainin_fall_2018/manuscript_data/matrices/rel_abundance_matrix/scaffolds/colonization/matrix_scaffolds_col_hp_uc.csv", row.names = 1)




treatment=c(rep("healthy", 6),rep("UC phage", 8)) 
colors=c(rep("blue", 6),rep("red", 8))


treatment.df.col.hpuc.scaf <-data.frame(treatment=c(rep("healthy",6),rep("UC phage",8)), my.data.col.hpuc.scaf)

treatment.df.col.hpuc.scaf

my.matrix.col.hpuc.scaf <- as.matrix(my.data.col.hpuc.scaf)

NMDS.col.hpuc.scaf <- metaMDS(my.matrix.col.hpuc.scaf, k=2)


ordiplot(NMDS.col.hpuc.scaf,type="points")
orditorp(NMDS.col.hpuc.scaf, display="sites")
#used orditorp to try to declutter NMDS

points(NMDS.col.hpuc.scaf, col=colors, pch=16, cex =2, data=my.data.col.hpuc.scaf)

text(NMDS.col.hpuc.scaf, cex =0.6, pos=4)


BC.col.scaffolds.hpuc.scaf <-vegdist(my.matrix.col.hpuc.scaf, method="bray", binary=FALSE, diag=FALSE, upper=FALSE, na.rm = FALSE)
adonis2(formula = BC.col.scaffolds.hpuc.scaf ~ treatment, data = treatment.df.col.hpuc.scaf , permutations = 999)

BC.col.scaffolds.hpuc.scaf <- set.seed(BC.col.scaffolds.hpuc.scaf)
#hp vs uc colonization scaffolds 
```



```{r}
#Comparing scaffold differences during phage gavage period between treatment groups

my.data.gavage.scaffolds <- read.csv("/Users/anshul.sinha/Desktop/Sequencing_analyses/phage_metagenomics_rainin_fall_2018/manuscript_data/matrices/rel_abundance_matrix/scaffolds/gavage/matrix_scaffolds_gav.csv", row.names = 1)


treatment=c(rep("PBS",7),rep("healthy", 9), rep("UC",9)) 
colors=c(rep("black",7),rep("blue",9), rep("red",9))


treatment.df.gavage.scaffolds <-data.frame(treatment=c(rep("PBS",7),rep("healthy",9),rep("UC",9)), my.data.gavage.scaffolds)

treatment.df.gavage.scaffolds

my.matrix.gavage.scaffolds <- as.matrix(my.data.gavage.scaffolds)

NMDS.gavage.scaffolds <- metaMDS(my.matrix.gavage.scaffolds, k=2)


ordiplot(NMDS.gavage.scaffolds,type="n", cex.axis = 2)

points(NMDS.gavage.scaffolds, col=colors, pch=16, cex=4)


#ordiellipse(NMDS.gavage.scaffolds,groups=treatment,col=c(rep("black",7),rep("blue",9),rep("red",9)),draw="polygon",alpha=0.4,border=c(rep("black",7),rep("blue",9),rep("red",9)), conf = 0.95)
```



```{r}
#Comparison of PBS control and UCP treatment during phage gavage period (scaffolds)

my.data.gav.npuc.scaf <- read.csv("/Users/anshul.sinha/Desktop/Sequencing_analyses/phage_metagenomics_rainin_fall_2018/manuscript_data/matrices/rel_abundance_matrix/scaffolds/gavage/matrix_scaffolds_gav_uc_np.csv", row.names = 1)


treatment=c(rep("no phage",7),rep("UC", 9)) 
colors=c(rep("black",7),rep("red",9))


treatment.df.gav.npuc.scaf <-data.frame(treatment=c(rep("no phage",7),rep("UC",9)), my.data.gav.npuc.scaf)

treatment.df.gav.npuc.scaf

my.matrix.gav.npuc.scaf <- as.matrix(my.data.gav.npuc.scaf)

NMDS.gav.npuc.scaf <- metaMDS(my.matrix.gav.npuc.scaf, k=2)



BC.gav.scaffolds.npuc.scaf <-vegdist(my.matrix.gav.npuc.scaf, method="bray", binary=FALSE, diag=FALSE, upper=FALSE, na.rm = FALSE)
adonis2(formula = BC.gav.scaffolds.npuc.scaf ~ treatment, data = treatment.df.gav.npuc.scaf , permutations = 999)

BC.gav.scaffolds.npuc.scaf <- set.seed(BC.gav.scaffolds.npuc.scaf)
#np vs uc gavage scaffolds 
```


```{r}
#Comparison of PBS control and HP treatment during phage gavage period (scaffolds)

my.data.gav.nphp.scaf <- read.csv("/Users/anshul.sinha/Desktop/Sequencing_analyses/phage_metagenomics_rainin_fall_2018/manuscript_data/matrices/rel_abundance_matrix/scaffolds/gavage/matrix_scaffolds_gav_np_hp.csv", row.names = 1)


treatment=c(rep("no phage",7),rep("healthy", 9)) 
colors=c(rep("black",7),rep("blue",9))


treatment.df.gav.nphp.scaf <-data.frame(treatment=c(rep("no phage",7),rep("healthy",9)), my.data.gav.nphp.scaf)

treatment.df.gav.nphp.scaf

my.matrix.gav.nphp.scaf <- as.matrix(my.data.gav.nphp.scaf)

NMDS.gav.nphp.scaf <- metaMDS(my.matrix.gav.nphp.scaf, k=2)


ordiplot(NMDS.gav.nphp.scaf,type="n")

points(NMDS.gav.nphp.scaf, col=colors, pch=16 )



BC.gav.scaffolds.nphp.scaf <-vegdist(my.matrix.gav.nphp.scaf, method="bray", binary=FALSE, diag=FALSE, upper=FALSE, na.rm = FALSE)
adonis2(formula = BC.gav.scaffolds.nphp.scaf ~ treatment, data = treatment.df.gav.nphp.scaf , permutations = 999)

BC.gav.scaffolds.nphp.scaf <- set.seed(BC.gav.scaffolds.nphp.scaf)
#np vs hp gavage scaffolds 

```

```{r}
#Comparison of HP and UCP treatments during phage gavage periods (scaffolds)
my.data.gav.hpuc.scaf <- read.csv("/Users/anshul.sinha/Desktop/Sequencing_analyses/phage_metagenomics_rainin_fall_2018/manuscript_data/matrices/rel_abundance_matrix/scaffolds/gavage/matrix_scaffolds_gav_hp_uc.csv", row.names = 1)




treatment=c(rep("healthy", 9),rep("UC phage", 9)) 
colors=c(rep("blue", 9),rep("red", 9))


treatment.df.gav.hpuc.scaf <-data.frame(treatment=c(rep("healthy",9),rep("UC phage",9)), my.data.gav.hpuc.scaf)

treatment.df.gav.hpuc.scaf

my.matrix.gav.hpuc.scaf <- as.matrix(my.data.gav.hpuc.scaf)

NMDS.gav.hpuc.scaf <- metaMDS(my.matrix.gav.hpuc.scaf, k=2)


ordiplot(NMDS.gav.hpuc.scaf,type="n")


points(NMDS.gav.hpuc.scaf, col=colors, pch=16)



BC.gav.scaffolds.hpuc.scaf <-vegdist(my.matrix.gav.hpuc.scaf, method="bray", binary=FALSE, diag=FALSE, upper=FALSE, na.rm = FALSE)
adonis2(formula = BC.gav.scaffolds.hpuc.scaf ~ treatment, data = treatment.df.gav.hpuc.scaf , permutations = 999)

BC.gav.scaffolds.hpuc.scaf <- set.seed(BC.gav.scaffolds.hpuc.scaf)
#hp vs uc gavage scaffolds 

```



```{r}

#Comparing scaffold differences during DSS washout period between treatment groups

my.data.dss.scaffolds <- read.csv("/Users/anshul.sinha/Desktop/Sequencing_analyses/phage_metagenomics_rainin_fall_2018/manuscript_data/matrices/rel_abundance_matrix/scaffolds/dss/matrix_scaffolds_dss.csv", row.names = 1)


treatment=c(rep("PBS",6),rep("healthy", 5), rep("UC",6)) 
colors=c(rep("black",6),rep("blue",5), rep("red",6))


treatment.df.dss.scaffolds <-data.frame(treatment=c(rep("PBS",6),rep("healthy",5),rep("UC",6)), my.data.dss.scaffolds)

treatment.df.dss.scaffolds

my.matrix.dss.scaffolds <- as.matrix(my.data.dss.scaffolds)

NMDS.dss.scaffolds <- metaMDS(my.matrix.dss.scaffolds, k=2)


ordiplot(NMDS.dss.scaffolds,type="n", cex.axis = 2)

points(NMDS.dss.scaffolds, col=colors, pch=16, cex =4)


```

```{r}
BC.dss.scaffolds <-vegdist(my.matrix.dss.scaffolds, method="bray", binary=FALSE, diag=FALSE, upper=FALSE, na.rm = FALSE)
adonis2(formula = BC.dss.scaffolds ~ treatment, data = treatment.df.dss.scaffolds , permutations = 999)

```


```{r}
#Comparison of PBS control and UCP treatment during DSS washout period (scaffold)
my.data.dss.npuc.scaf <- read.csv("/Users/anshul.sinha/Desktop/Sequencing_analyses/phage_metagenomics_rainin_fall_2018/manuscript_data/matrices/rel_abundance_matrix/scaffolds/dss/matrix_scaffolds_dss_np_uc.csv", row.names = 1)


treatment=c(rep("no phage",6),rep("UC", 6)) 
colors=c(rep("black",6),rep("red",6))


treatment.df.dss.npuc.scaf <-data.frame(treatment=c(rep("no phage",6),rep("UC",6)), my.data.dss.npuc.scaf)

treatment.df.dss.npuc.scaf

my.matrix.dss.npuc.scaf <- as.matrix(my.data.dss.npuc.scaf)

NMDS.dss.npuc.scaf <- metaMDS(my.matrix.dss.npuc.scaf, k=2)



BC.dss.scaffolds.npuc.scaf <-vegdist(my.matrix.dss.npuc.scaf, method="bray", binary=FALSE, diag=FALSE, upper=FALSE, na.rm = FALSE)
adonis2(formula = BC.dss.scaffolds.npuc.scaf ~ treatment, data = treatment.df.dss.npuc.scaf , permutations = 999)

BC.dss.scaffolds.npuc.scaf <- set.seed(BC.dss.scaffolds.npuc.scaf)
#np vs uc dss scaffolds 
```

```{r}

#Comparison of PBS control and HP treatment during DSS washout period (scaffolds)
my.data.dss.nphp.scaf <- read.csv("/Users/anshul.sinha/Desktop/Sequencing_analyses/phage_metagenomics_rainin_fall_2018/manuscript_data/matrices/rel_abundance_matrix/scaffolds/dss/matrix_scaffolds_np_hp_dss.csv", row.names = 1)


treatment=c(rep("no phage",6),rep("healthy", 5)) 
colors=c(rep("black",6),rep("blue",5))


treatment.df.dss.nphp.scaf <-data.frame(treatment=c(rep("no phage",6),rep("healthy",5)), my.data.dss.nphp.scaf)

treatment.df.dss.nphp.scaf

my.matrix.dss.nphp.scaf <- as.matrix(my.data.dss.nphp.scaf)

NMDS.dss.nphp.scaf <- metaMDS(my.matrix.dss.nphp.scaf, k=2)


ordiplot(NMDS.dss.nphp.scaf,type="n")

points(NMDS.dss.nphp.scaf, col=colors, pch=16 )



BC.dss.scaffolds.nphp.scaf <-vegdist(my.matrix.dss.nphp.scaf, method="bray", binary=FALSE, diag=FALSE, upper=FALSE, na.rm = FALSE)
adonis2(formula = BC.dss.scaffolds.nphp.scaf ~ treatment, data = treatment.df.dss.nphp.scaf , permutations = 999)

BC.dss.scaffolds.nphp.scaf <- set.seed(BC.dss.scaffolds.nphp.scaf)
#np vs hp gavage scaffolds 

```

```{r}
#Comparison of HP and UCP treatments during DSS washout period (scaffolds)

my.data.dss.hpuc.scaf <- read.csv("/Users/anshul.sinha/Desktop/Sequencing_analyses/phage_metagenomics_rainin_fall_2018/manuscript_data/matrices/rel_abundance_matrix/scaffolds/dss/matrix_scaffolds_dss_hp_uc.csv", row.names = 1)




treatment=c(rep("healthy", 5),rep("UC phage", 6)) 
colors=c(rep("blue", 5),rep("red", 6))


treatment.df.dss.hpuc.scaf <-data.frame(treatment=c(rep("healthy",5),rep("UC phage",6)), my.data.dss.hpuc.scaf)

treatment.df.dss.hpuc.scaf

my.matrix.dss.hpuc.scaf <- as.matrix(my.data.dss.hpuc.scaf)

NMDS.dss.hpuc.scaf <- metaMDS(my.matrix.dss.hpuc.scaf, k=2)


ordiplot(NMDS.dss.hpuc.scaf,type="n")


points(NMDS.dss.hpuc.scaf, col=colors, pch=16)



BC.dss.scaffolds.hpuc.scaf <-vegdist(my.matrix.dss.hpuc.scaf, method="bray", binary=FALSE, diag=FALSE, upper=FALSE, na.rm = FALSE)
adonis2(formula = BC.dss.scaffolds.hpuc.scaf ~ treatment, data = treatment.df.dss.hpuc.scaf , permutations = 999)

BC.dss.scaffolds.hpuc.scaf <- set.seed(BC.dss.scaffolds.hpuc.scaf)
#hp vs uc gavage scaffolds 

```

