---
title: "deseq_dss"
output: html_document
---




```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

```{r}
R.version.string
```

```{r}
install.packages("ggrepel")
```

```{r}
library("ggplot2") 
library("ggrepel") 
library(DESeq2)
library("dplyr")
library(ggrepel)
library(tidyverse)
```


```{r}
library(EnhancedVolcano)
```

```{r}

#input is raw counts matrix from samples from the DSS washout period 


raw_counts_dss <- read.csv("/Users/anshul.sinha/Desktop/Sequencing_analyses/phage_metagenomics_rainin_fall_2018/manuscript_data/deseq/data/mouse_only/dss/matrix_read_counts_dss.csv", header = TRUE)

raw_counts_dss

#switching the columns and rows so that each row is a different contig and each column is a different sample 
switch_matrix_dss <-  t(raw_counts_dss)



switch_matrix_dss

write.csv(switch_matrix_dss, "/Users/anshul.sinha/Desktop/Sequencing_analyses/phage_metagenomics_rainin_fall_2018/manuscript_data/deseq/data/mouse_only/dss/switched_matrix_dss.csv") 


#exporting csv here (contains top row with v1-73, will remove manually in csv file)

```



```{r}
#reading in manually edited switched matrix
raw_counts2_dss <- read.csv("/Users/anshul.sinha/Desktop/Sequencing_analyses/phage_metagenomics_rainin_fall_2018/manuscript_data/deseq/data/mouse_only/dss/switched_matrix_dss.csv", check.names = FALSE)
raw_counts2_dss
```



```{r}


colnames(raw_counts2_dss)[1] <- "contig_ID"

raw_counts2_dss



#naming that column "contig_ID"

contig_ID_dss <-  raw_counts2_dss$contig_ID

#Now I am generating a "sample index" so I can exclude "contig_ID" from our matrix 

sampleindex_dss <- grepl("\\d", colnames(raw_counts2_dss))
#sampleindex
#converting to a matrix and only including the sample index columns 
raw_counts_matrix_dss <- as.matrix(raw_counts2_dss[,sampleindex_dss])
raw_counts_matrix_dss


rownames(raw_counts_matrix_dss) <- contig_ID_dss

head(raw_counts_matrix_dss)

write.csv(raw_counts_matrix_dss, "/Users/anshul.sinha/Desktop/Sequencing_analyses/phage_metagenomics_rainin_fall_2018/manuscript_data/deseq/data/mouse_only/dss/raw_counts_matrix_dss.csv")





```


```{r}
sample_mapping_dss <- read.csv("/Users/anshul.sinha/Desktop/Sequencing_analyses/phage_metagenomics_rainin_fall_2018/manuscript_data/deseq/data/mouse_only/dss/sample_mapping_deseq_dss.csv")
```



```{r}
#Now need to change the sample mapping data so that it can be used by DEseq2
sample_mapping_dss
rownames(sample_mapping_dss) <- sample_mapping_dss$Sample



#for now, I am only concerned with treatment differences and Sample Period differences, so I will just keep those columns 
keep <- c("Treatment", "Period")

sample_mapping_dss <- sample_mapping_dss[,keep]

colnames(sample_mapping_dss) <- c("Treatment", "Period")

sample_mapping_dss <- as.matrix(sample_mapping_dss)

colnames(raw_counts_matrix_dss)
row.names(sample_mapping_dss)

#Put the columns of the count data in the same order as rows names of the sample mapping, then make sure it worked (TRUE)

raw_counts_matrix_dss <- raw_counts_matrix_dss[, unique(rownames(sample_mapping_dss))]


all(colnames(raw_counts_matrix_dss) == rownames(sample_mapping_dss))

sample_mapping_dss
raw_counts_matrix_dss

#True if alignment is correct



```


```{r}
#creating deseq2 data object 
deseq2Data_dss <- DESeqDataSetFromMatrix(countData=raw_counts_matrix_dss, colData=sample_mapping_dss, design= ~  Treatment)
```


```{r}
deseq2Data_dss <- estimateSizeFactors(deseq2Data_dss, type= "poscounts")

deseq2Data_dss <- DESeq(deseq2Data_dss)
```



```{r}
resultsUCHP_dss <- results(deseq2Data_dss, contrast = c("Treatment", "UCP", "HP"),alpha = 0.001, lfcThreshold = 1)
summary(resultsUCHP_dss)

resultsUCNP_dss <- results(deseq2Data_dss, contrast = c("Treatment", "UCP", "NP"), alpha = 0.001, lfcThreshold = 1)
summary(resultsUCNP_dss)

resultsHPNP_dss <- results(deseq2Data_dss, contrast = c("Treatment", "HP", "NP"), alpha = 0.001, lfcThreshold = 1)
summary(resultsHPNP_dss)
                                    
                       
```

```{r}
contig_metadata <- read.csv("/Users/anshul.sinha/Desktop/Sequencing_analyses/phage_metagenomics_rainin_fall_2018/manuscript_data/deseq/data/metadata_updated.csv")

contig_metadata
```

```{r}

#healthy phage vs. PBS comparison 

sig_results_HPNP_dss <-  subset(resultsHPNP_dss, padj < 0.001)
#now only taking only the values that meet the p-value threshold

df_HPNP_dss <- as.data.frame(sig_results_HPNP_dss)
df_HPNP_dss <- df_HPNP_dss %>% 
  rownames_to_column(var = "ContigName")
#converting results table and naming first column so results table can be annotated



df_HPNP_ordered_dss <- df_HPNP_dss[order(df_HPNP_dss$pvalue),]




annotated_HPNP_dss <- left_join(df_HPNP_ordered_dss, contig_metadata, by=c("ContigName"="ContigName"))

annotated_HPNP_dss



write_csv(annotated_HPNP_dss, "/Users/anshul.sinha/Desktop/Sequencing_analyses/phage_metagenomics_rainin_fall_2018/manuscript_data/deseq/data/mouse_only/dss/unformatted_outputs/HPNP.csv")

annotated_HPNP_dss_topcrispr <- subset(annotated_HPNP_dss, !is.na(annotated_HPNP_dss$CRISPRHost))

annotated_HPNP_dss_topcrispr

write_csv(annotated_HPNP_dss_topcrispr, "/Users/anshul.sinha/Desktop/Sequencing_analyses/phage_metagenomics_rainin_fall_2018/manuscript_data/deseq/data/mouse_only/dss/HPNP_crispr.csv")


annotated_HPNP_dss_AMGs <- subset(annotated_HPNP_dss, !is.na(annotated_HPNP_dss$AMG))

annotated_HPNP_dss_AMGs

write_csv(annotated_HPNP_dss_AMGs, "/Users/anshul.sinha/Desktop/Sequencing_analyses/phage_metagenomics_rainin_fall_2018/manuscript_data/deseq/data/mouse_only/dss/AMG/HPNP_dss_amgs.csv")



```




```{r}
#Counting numbers of each AMG upregulated in each comparison and plotting using a pie chart

subset_dss_HPNP_AMGs <-  subset(annotated_HPNP_dss_AMGs, log2FoldChange > 0)

subset_dss_HPNP_AMGs_split <- strsplit(subset_dss_HPNP_AMGs$AMG, split = "++", fixed = TRUE)


subset_dss_HPNP_AMGs_split_df <- data_frame(v1= rep(subset_dss_HPNP_AMGs$ContigName, sapply(subset_dss_HPNP_AMGs_split, length)), v2= unlist(subset_dss_HPNP_AMGs_split))

subset_dss_HPNP_AMGs_split_df

table_AMGs_HPNP_AMGs_dss <- table(subset_dss_HPNP_AMGs_split_df$v2) 
df_AMG_freq_HPNP_dss <- as.data.frame(table_AMGs_HPNP_AMGs_dss)

colnames(df_AMG_freq_HPNP_dss)[1] <- "AMG"

df_AMG_freq_HPNP_dss

write.csv(df_AMG_freq_HPNP_dss, "/Users/anshul.sinha/Desktop/Sequencing_analyses/phage_metagenomics_rainin_fall_2018/manuscript_data/deseq/data/mouse_only/dss/AMG/num_amgs_upregulated_hpnp.csv" )

pie_AMGs_HPNP <- ggplot(df_AMG_freq_HPNP_dss, aes(x="", y=Freq, fill=AMG))+
geom_bar(width = 1, stat = "identity", color= "white") + 
coord_polar("y", start=0) +
theme_void() 
pie_AMGs_HPNP
```




```{r}
#UC phage vs. PBS comparison, ordering by adjusted p-val and then filtering for those with a CRISPR match

sig_results_UCNP_dss <-  subset(resultsUCNP_dss, padj < 0.001)
#now only taking only the values that meet the p-value threshold

df_UCNP_dss <- as.data.frame(sig_results_UCNP_dss)
df_UCNP_dss <- df_UCNP_dss %>% 
  rownames_to_column(var = "ContigName")
#converting results table and naming first column so results table can be annotated



df_UCNP_ordered_dss <- df_UCNP_dss[order(df_UCNP_dss$pvalue),]
#ordering by p-adjusted



annotated_UCNP_dss <- left_join(df_UCNP_ordered_dss, contig_metadata, by=c("ContigName"="ContigName"))

annotated_UCNP_dss



write_csv(annotated_UCNP_dss, "/Users/anshul.sinha/Desktop/Sequencing_analyses/phage_metagenomics_rainin_fall_2018/manuscript_data/deseq/data/mouse_only/dss/unformatted_outputs/UCNP.csv")


annotated_UCNP_dss_topcrispr <- subset(annotated_UCNP_dss, !is.na(annotated_UCNP_dss$CRISPRHost))

annotated_UCNP_dss_topcrispr

write_csv(annotated_UCNP_dss_topcrispr, "/Users/anshul.sinha/Desktop/Sequencing_analyses/phage_metagenomics_rainin_fall_2018/manuscript_data/deseq/data/mouse_only/dss/UCNP_crispr.csv")


annotated_UCNP_dss_AMGs <- subset(annotated_UCNP_dss, !is.na(annotated_UCNP_dss$AMG))

annotated_UCNP_dss_AMGs

write_csv(annotated_UCNP_dss_AMGs, "/Users/anshul.sinha/Desktop/Sequencing_analyses/phage_metagenomics_rainin_fall_2018/manuscript_data/deseq/data/mouse_only/dss/AMG/UCNP_dss_amgs.csv")


```

```{r}
#Counting numbers of each AMG upregulated in each comparison and plotting using a pie chart

subset_dss_UCNP_AMGs <-  subset(annotated_UCNP_dss_AMGs, log2FoldChange > 0)

subset_dss_UCNP_AMGs_split <- strsplit(subset_dss_UCNP_AMGs$AMG, split = "++", fixed = TRUE)


subset_dss_UCNP_AMGs_split_df <- data_frame(v1= rep(subset_dss_UCNP_AMGs$ContigName, sapply(subset_dss_UCNP_AMGs_split, length)), v2= unlist(subset_dss_UCNP_AMGs_split))

subset_dss_UCNP_AMGs_split_df

table_AMGs_UCNP_AMGs_dss <- table(subset_dss_UCNP_AMGs_split_df$v2) 
df_AMG_freq_UCNP_dss <- as.data.frame(table_AMGs_UCNP_AMGs_dss)

colnames(df_AMG_freq_UCNP_dss)[1] <- "AMG"

df_AMG_freq_UCNP_dss

write.csv(df_AMG_freq_UCNP_dss, "/Users/anshul.sinha/Desktop/Sequencing_analyses/phage_metagenomics_rainin_fall_2018/manuscript_data/deseq/data/mouse_only/dss/AMG/num_amgs_upregulated_ucnp.csv" )

pie_AMGs_UCNP <- ggplot(df_AMG_freq_UCNP_dss, aes(x="", y=Freq, fill=AMG))+
geom_bar(width = 1, stat = "identity", color= "white") + 
coord_polar("y", start=0) +
theme_void() 
pie_AMGs_UCNP
```




```{r}
#UC phage vs. HP comparison 

sig_results_UCHP_dss <-  subset(resultsUCHP_dss, padj < 0.001)
#now only taking only the values that meet the p-value threshold

df_UCHP_dss <- as.data.frame(sig_results_UCHP_dss)
df_UCHP_dss <- df_UCHP_dss %>% 
  rownames_to_column(var = "ContigName")
#converting results table and naming first column so results table can be annotated



df_UCHP_ordered_dss <- df_UCHP_dss[order(df_UCHP_dss$pvalue),]




annotated_UCHP_dss <- left_join(df_UCHP_ordered_dss, contig_metadata, by=c("ContigName"="ContigName"))

annotated_UCHP_dss



write_csv(annotated_UCHP_dss, "/Users/anshul.sinha/Desktop/Sequencing_analyses/phage_metagenomics_rainin_fall_2018/manuscript_data/deseq/data/mouse_only/dss/unformatted_outputs/UCHP.csv")

annotated_UCHP_dss_topcrispr <- subset(annotated_UCHP_dss, !is.na(annotated_UCHP_dss$CRISPRHost))

annotated_UCHP_dss_topcrispr

write_csv(annotated_UCHP_dss_topcrispr, "/Users/anshul.sinha/Desktop/Sequencing_analyses/phage_metagenomics_rainin_fall_2018/manuscript_data/deseq/data/mouse_only/dss/UCHP_crispr.csv")


annotated_UCHP_dss_AMGs <- subset(annotated_UCHP_dss, !is.na(annotated_UCHP_dss$AMG))

annotated_UCHP_dss_AMGs


write_csv(annotated_UCHP_dss_AMGs, "/Users/anshul.sinha/Desktop/Sequencing_analyses/phage_metagenomics_rainin_fall_2018/manuscript_data/deseq/data/mouse_only/dss/AMG/UCHP_dss_amgs.csv")

```

```{r}
#Counting numbers of each AMG upregulated in each comparison and plotting using a pie chart

subset_dss_UCHP_AMGs <-  subset(annotated_UCHP_dss_AMGs, log2FoldChange > 0)

subset_dss_UCHP_AMGs_split <- strsplit(subset_dss_UCHP_AMGs$AMG, split = "++", fixed = TRUE)


subset_dss_UCHP_AMGs_split_df <- data_frame(v1= rep(subset_dss_UCHP_AMGs$ContigName, sapply(subset_dss_UCHP_AMGs_split, length)), v2= unlist(subset_dss_UCHP_AMGs_split))

subset_dss_UCHP_AMGs_split_df

table_AMGs_UCHP_AMGs_dss <- table(subset_dss_UCHP_AMGs_split_df$v2) 
df_AMG_freq_UCHP_dss <- as.data.frame(table_AMGs_UCHP_AMGs_dss)

colnames(df_AMG_freq_UCHP_dss)[1] <- "AMG"

df_AMG_freq_UCHP_dss

write.csv(df_AMG_freq_UCHP_dss, "/Users/anshul.sinha/Desktop/Sequencing_analyses/phage_metagenomics_rainin_fall_2018/manuscript_data/deseq/data/mouse_only/dss/AMG/num_amgs_upregulated_hpnp.csv" )

pie_AMGs_UCHP <- ggplot(df_AMG_freq_UCHP_dss, aes(x="", y=Freq, fill=AMG))+
geom_bar(width = 1, stat = "identity", color= "white") + 
coord_polar("y", start=0) +
theme_void() 
pie_AMGs_UCHP
```






```{r}
#volcano plot- UC VLPs vs. PBS control 

EV_dss_UCNP <- EnhancedVolcano(resultsUCNP_dss, lab=NA , x = 'log2FoldChange',
    y = 'pvalue', pCutoff = 0.001, pointSize = 2)

EV_dss_UCNP



ggsave("/Users/anshul.sinha/Desktop/Sequencing_analyses/phage_metagenomics_rainin_fall_2018/manuscript_data/deseq/data/mouse_only/dss/volcano_plots/UCPBS_dss_volcano.tiff", width =7 , height = 8, dpi=300)


```

```{r}
#volcano plot- HP VLPs vs. PBS control 

EV_dss_HPNP <- EnhancedVolcano(resultsHPNP_dss, lab=NA , x = 'log2FoldChange',
    y = 'pvalue', pCutoff = 0.001, pointSize = 2)

EV_dss_HPNP

ggsave("/Users/anshul.sinha/Desktop/Sequencing_analyses/phage_metagenomics_rainin_fall_2018/manuscript_data/deseq/data/mouse_only/dss/volcano_plots/HPBPS_volcano.tiff", width =7 , height = 8, dpi=300)

```


```{r}
#volcano plot- UC VLPs vs. HP VLPs  

EV_dss_UCHP <- EnhancedVolcano(resultsUCHP_dss, lab=NA , x = 'log2FoldChange',
    y = 'pvalue', pCutoff = 0.001, pointSize = 2)

EV_dss_UCHP

```


