---
title: "deseq_gav"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```




```{r}
library(DESeq2)
library(dplyr)
library(tidyverse)
library(EnhancedVolcano)
```


```{r}
#Input is raw counts matrix including samples from VLP gavage period

raw_counts_gav <- read.csv("/Users/anshul.sinha/Desktop/Sequencing_analyses/phage_metagenomics_rainin_fall_2018/manuscript_data/deseq/data/mouse_only/gavage/matrix_read_counts_gavage.csv", header = TRUE)

raw_counts_gav

#switching the columns and rows so that each row is a different contig and each column is a different sample 
switch_matrix_gav <-  t(raw_counts_gav)



switch_matrix_gav

write.csv(switch_matrix_gav, "/Users/anshul.sinha/Desktop/Sequencing_analyses/phage_metagenomics_rainin_fall_2018/manuscript_data/deseq/data/mouse_only/gavage/switched_matrix_gav.csv") 



```



```{r}
#reading in manually edited switched matrix
raw_counts2_gav <- read.csv("/Users/anshul.sinha/Desktop/Sequencing_analyses/phage_metagenomics_rainin_fall_2018/manuscript_data/deseq/data/mouse_only/gavage/switched_matrix_gav.csv", check.names = FALSE)
raw_counts2_gav
```



```{r}


colnames(raw_counts2_gav)[1] <- "contig_ID"

raw_counts2_gav



#naming that column "contig_ID"

contig_ID_gav <-  raw_counts2_gav$contig_ID

#Now I am generating a "sample index" so I can exclude "contig_ID" from our matrix 

sampleindex_gav <- grepl("\\d", colnames(raw_counts2_gav))
#sampleindex
#converting to a matrix and only including the sample index columns 
raw_counts_matrix_gav <- as.matrix(raw_counts2_gav[,sampleindex_gav])
raw_counts_matrix_gav


rownames(raw_counts_matrix_gav) <- contig_ID_gav

head(raw_counts_matrix_gav)

write.csv(raw_counts_matrix_gav, "/Users/anshul.sinha/Desktop/Sequencing_analyses/phage_metagenomics_rainin_fall_2018/manuscript_data/deseq/data/mouse_only/gavage/raw_counts_matrix_gav.csv")





```


```{r}
sample_mapping_gav <- read.csv("/Users/anshul.sinha/Desktop/Sequencing_analyses/phage_metagenomics_rainin_fall_2018/manuscript_data/deseq/data/mouse_only/gavage/sample_mapping_deseq_gavage.csv")
```



```{r}
#Now need to change the sample mapping data so that it can be used by DEseq2
sample_mapping_gav
rownames(sample_mapping_gav) <- sample_mapping_gav$Sample



#for now, I am only concerned with treatment differences and Sample Period differences, so I will just keep those columns 
keep <- c("Treatment", "Period")

sample_mapping_gav <- sample_mapping_gav[,keep]

colnames(sample_mapping_gav) <- c("Treatment", "Period")

sample_mapping_gav <- as.matrix(sample_mapping_gav)

colnames(raw_counts_matrix_gav)
row.names(sample_mapping_gav)

#Put the columns of the count data in the same order as rows names of the sample mapping, then make sure it worked (TRUE)

raw_counts_matrix_gav <- raw_counts_matrix_gav[, unique(rownames(sample_mapping_gav))]


all(colnames(raw_counts_matrix_gav) == rownames(sample_mapping_gav))

sample_mapping_gav
raw_counts_matrix_gav

#True if alignment is correct



```


```{r}
#creating deseq2 data object 
deseq2Data_gav <- DESeqDataSetFromMatrix(countData=raw_counts_matrix_gav, colData=sample_mapping_gav, design= ~  Treatment)
```


```{r}
deseq2Data_gav <- estimateSizeFactors(deseq2Data_gav, type= "poscounts")

deseq2Data_gav <- DESeq(deseq2Data_gav)
```



```{r}
resultsUCHP_gav <- results(deseq2Data_gav, contrast = c("Treatment", "UCP", "HP"), alpha = 0.001, lfcThreshold = 1)
summary(resultsUCHP_gav)

resultsUCNP_gav <- results(deseq2Data_gav, contrast = c("Treatment", "UCP", "NP"), alpha = 0.001, lfcThreshold = 1)
summary(resultsUCNP_gav)

resultsHPNP_gav <- results(deseq2Data_gav, contrast = c("Treatment", "HP", "NP"),alpha = 0.001, lfcThreshold = 1)
summary(resultsHPNP_gav)

                                    
                       
```

```{r}
contig_metadata <- read.csv("/Users/anshul.sinha/Desktop/Sequencing_analyses/phage_metagenomics_rainin_fall_2018/manuscript_data/deseq/data/metadata_updated.csv")

contig_metadata

```






```{r}
#healthy phage vs. PBS comparison 

sig_results_HPNP_gav <-  subset(resultsHPNP_gav, padj < 0.001)
#now only taking only the values that meet the p-value threshold

df_HPNP_gav <- as.data.frame(sig_results_HPNP_gav)
df_HPNP_gav <- df_HPNP_gav %>% 
  rownames_to_column(var = "ContigName")
#converting results table and naming first column so results table can be annotated



df_HPNP_ordered_gav <- df_HPNP_gav[order(df_HPNP_gav$pvalue),]




annotated_HPNP_gav <- left_join(df_HPNP_ordered_gav, contig_metadata, by=c("ContigName"="ContigName"))

annotated_HPNP_gav



write_csv(annotated_HPNP_gav, "/Users/anshul.sinha/Desktop/Sequencing_analyses/phage_metagenomics_rainin_fall_2018/manuscript_data/deseq/data/mouse_only/gavage/outputs_unformatted/HPNP.csv")

annotated_HPNP_gav_topcrispr <- subset(annotated_HPNP_gav, !is.na(annotated_HPNP_gav$CRISPRHost))

annotated_HPNP_gav_topcrispr

write_csv(annotated_HPNP_gav_topcrispr, "/Users/anshul.sinha/Desktop/Sequencing_analyses/phage_metagenomics_rainin_fall_2018/manuscript_data/deseq/data/mouse_only/gavage/HPNP_crispr.csv")


annotated_HPNP_gav_AMGs <- subset(annotated_HPNP_gav, !is.na(annotated_HPNP_gav$AMG))

annotated_HPNP_gav_AMGs

write_csv(annotated_HPNP_gav_AMGs, "/Users/anshul.sinha/Desktop/Sequencing_analyses/phage_metagenomics_rainin_fall_2018/manuscript_data/deseq/data/mouse_only/gavage/AMGs/HPNP_gav_AMGs.csv")


```

```{r}
#Counting numbers of each AMG upregulated in each comparison and plotting using a pie chart

subset_gav_HPNP_AMGs <-  subset(annotated_HPNP_gav_AMGs, log2FoldChange > 0)

subset_gav_HPNP_AMGs_split <- strsplit(subset_gav_HPNP_AMGs$AMG, split = "++", fixed = TRUE)


subset_gav_HPNP_AMGs_split_df <- data_frame(v1= rep(subset_gav_HPNP_AMGs$ContigName, sapply(subset_gav_HPNP_AMGs_split, length)), v2= unlist(subset_gav_HPNP_AMGs_split))

subset_gav_HPNP_AMGs_split_df

table_AMGs_HPNP_AMGs_gav <- table(subset_gav_HPNP_AMGs_split_df$v2) 
df_AMG_freq_HPNP_gav <- as.data.frame(table_AMGs_HPNP_AMGs_gav)

colnames(df_AMG_freq_HPNP_gav)[1] <- "AMG"

df_AMG_freq_HPNP_gav

write.csv(df_AMG_freq_HPNP_gav, "/Users/anshul.sinha/Desktop/Sequencing_analyses/phage_metagenomics_rainin_fall_2018/manuscript_data/deseq/data/mouse_only/gavage/HPNP_upregulated_AMG.csv" )

pie_AMGs_HPNP_gav <- ggplot(df_AMG_freq_HPNP_gav, aes(x="", y=Freq, fill=AMG))+
geom_bar(width = 1, stat = "identity", color= "white") + 
coord_polar("y", start=0) +
theme_void() 
pie_AMGs_HPNP_gav
```


```{r}
#UC phage vs. PBS comparison 

sig_results_UCNP_gav <-  subset(resultsUCNP_gav, padj < 0.001)
#now only taking only the values that meet the p-value threshold

df_UCNP_gav <- as.data.frame(sig_results_UCNP_gav)
df_UCNP_gav <- df_UCNP_gav %>% 
  rownames_to_column(var = "ContigName")
#converting results table and naming first column so results table can be annotated



df_UCNP_ordered_gav <- df_UCNP_gav[order(df_UCNP_gav$pvalue),]




annotated_UCNP_gav <- left_join(df_UCNP_ordered_gav, contig_metadata, by=c("ContigName"="ContigName"))

annotated_UCNP_gav



write_csv(annotated_UCNP_gav, "/Users/anshul.sinha/Desktop/Sequencing_analyses/phage_metagenomics_rainin_fall_2018/manuscript_data/deseq/data/mouse_only/gavage/outputs_unformatted/UCNP.csv")

annotated_UCNP_gav_topcrispr <- subset(annotated_UCNP_gav, !is.na(annotated_UCNP_gav$CRISPRHost))

annotated_UCNP_gav_topcrispr

write_csv(annotated_UCNP_gav_topcrispr, "/Users/anshul.sinha/Desktop/Sequencing_analyses/phage_metagenomics_rainin_fall_2018/manuscript_data/deseq/data/mouse_only/gavage/UCNP_crispr.csv")

annotated_UCNP_gav_AMGs <- subset(annotated_UCNP_gav, !is.na(annotated_UCNP_gav$AMG))

annotated_UCNP_gav_AMGs

write_csv(annotated_UCNP_gav_AMGs, "/Users/anshul.sinha/Desktop/Sequencing_analyses/phage_metagenomics_rainin_fall_2018/manuscript_data/deseq/data/mouse_only/gavage/AMGs/UCNP_gav_AMGs.csv")


```


```{r}
#Counting numbers of each AMG upregulated in each comparison and plotting using a pie chart

subset_gav_UCNP_AMGs <-  subset(annotated_UCNP_gav_AMGs, log2FoldChange > 0)

subset_gav_UCNP_AMGs_split <- strsplit(subset_gav_UCNP_AMGs$AMG, split = "++", fixed = TRUE)


subset_gav_UCNP_AMGs_split_df <- data_frame(v1= rep(subset_gav_UCNP_AMGs$ContigName, sapply(subset_gav_UCNP_AMGs_split, length)), v2= unlist(subset_gav_UCNP_AMGs_split))

subset_gav_UCNP_AMGs_split_df

table_AMGs_UCNP_AMGs_gav <- table(subset_gav_UCNP_AMGs_split_df$v2) 
df_AMG_freq_UCNP_gav <- as.data.frame(table_AMGs_UCNP_AMGs_gav)

colnames(df_AMG_freq_UCNP_gav)[1] <- "AMG"

df_AMG_freq_UCNP_gav

write.csv(df_AMG_freq_UCNP_gav, "/Users/anshul.sinha/Desktop/Sequencing_analyses/phage_metagenomics_rainin_fall_2018/manuscript_data/deseq/data/mouse_only/gavage/UCNP_upregulated_AMG.csv" )

pie_AMGs_UCNP_gav <- ggplot(df_AMG_freq_UCNP_gav, aes(x="", y=Freq, fill=AMG))+
geom_bar(width = 1, stat = "identity", color= "white") + 
coord_polar("y", start=0) +
theme_void() 
pie_AMGs_UCNP_gav
```





```{r}
#UC phage vs. HP comparison 

sig_results_UCHP_gav <-  subset(resultsUCHP_gav, padj < 0.001)
#now only taking only the values that meet the p-value threshold

df_UCHP_gav <- as.data.frame(sig_results_UCHP_gav)
df_UCHP_gav <- df_UCHP_gav %>% 
  rownames_to_column(var = "ContigName")
#converting results table and naming first column so results table can be annotated



df_UCHP_ordered_gav <- df_UCHP_gav[order(df_UCHP_gav$pvalue),]




annotated_UCHP_gav <- left_join(df_UCHP_ordered_gav, contig_metadata, by=c("ContigName"="ContigName"))

annotated_UCHP_gav



write_csv(annotated_UCHP_gav, "/Users/anshul.sinha/Desktop/Sequencing_analyses/phage_metagenomics_rainin_fall_2018/manuscript_data/deseq/data/mouse_only/gavage/outputs_unformatted/UCHP.csv")

annotated_UCHP_gav_topcrispr <- subset(annotated_UCHP_gav, !is.na(annotated_UCHP_gav$CRISPRHost))

annotated_UCHP_gav_topcrispr

write_csv(annotated_UCHP_gav_topcrispr, "/Users/anshul.sinha/Desktop/Sequencing_analyses/phage_metagenomics_rainin_fall_2018/manuscript_data/deseq/data/mouse_only/gavage/UCHP_crispr.csv")

annotated_UCHP_gav_AMGs <- subset(annotated_UCHP_gav, !is.na(annotated_UCHP_gav$AMG))

annotated_UCHP_gav_AMGs

write_csv(annotated_UCHP_gav_AMGs, "/Users/anshul.sinha/Desktop/Sequencing_analyses/phage_metagenomics_rainin_fall_2018/manuscript_data/deseq/data/mouse_only/gavage/AMGs/UCHP_gav_AMGs.csv")

```


```{r}
#Counting numbers of each AMG upregulated in each comparison and plotting using a pie chart

subset_gav_UCHP_AMGs <-  subset(annotated_UCHP_gav_AMGs, log2FoldChange > 0)

subset_gav_UCHP_AMGs_split <- strsplit(subset_gav_UCHP_AMGs$AMG, split = "++", fixed = TRUE)


subset_gav_UCHP_AMGs_split_df <- data_frame(v1= rep(subset_gav_UCHP_AMGs$ContigName, sapply(subset_gav_UCHP_AMGs_split, length)), v2= unlist(subset_gav_UCHP_AMGs_split))

subset_gav_UCHP_AMGs_split_df

table_AMGs_UCHP_AMGs_gav <- table(subset_gav_UCHP_AMGs_split_df$v2) 
df_AMG_freq_UCHP_gav <- as.data.frame(table_AMGs_UCHP_AMGs_gav)

colnames(df_AMG_freq_UCHP_gav)[1] <- "AMG"

df_AMG_freq_UCHP_gav

write.csv(df_AMG_freq_UCHP_gav, "/Users/anshul.sinha/Desktop/Sequencing_analyses/phage_metagenomics_rainin_fall_2018/manuscript_data/deseq/data/mouse_only/gavage/UCHP_upregulated_AMG.csv" )

pie_AMGs_UCHP_gav <- ggplot(df_AMG_freq_UCHP_gav, aes(x="", y=Freq, fill=AMG))+
geom_bar(width = 1, stat = "identity", color= "white") + 
coord_polar("y", start=0) +
theme_void() 
pie_AMGs_UCHP_gav
```


```{r}
#volcano plot - VLP gavage period: UC vs. PBS control comparison

keyvals <- ifelse(
    resultsUCNP_gav$log2FoldChange < -1 & resultsUCNP_gav$padj <0.001, 'black',
      ifelse(resultsUCNP_gav$log2FoldChange > 1 & resultsUCNP_gav$padj <0.001, 'red',
        'light grey'))
  keyvals[is.na(keyvals)] <- 'light grey'
  names(keyvals)[keyvals == 'red'] <- 'Over-Abundant UC VLP Scaffolds'
  names(keyvals)[keyvals == 'light grey'] <- 'Non-significant'
  names(keyvals)[keyvals == 'black'] <- 'Under-Abundant UC VLP Scaffolds'

EV_gav_UCNP <- EnhancedVolcano(resultsUCNP_gav, lab=NA , x = 'log2FoldChange',
    y = 'pvalue', pCutoff = 0.001, selectLab = rownames(resultsUCNP_gav)[which(names(keyvals) %in% c('Over-Abundant UC VLP Scaffolds', 'Under-Abundant UC VLP Scaffolds'))], pointSize = 2, gridlines.major = FALSE, gridlines.minor = FALSE, colCustom = keyvals,  legendPosition = 'right', legendLabSize = 12,legendIconSize = 4.0, title = "UCbac + UC VLPs vs. UCbac + PBS",     subtitle = "", caption = ""
)

EV_gav_UCNP


ggsave("/Users/anshul.sinha/Desktop/Sequencing_analyses/phage_metagenomics_rainin_fall_2018/manuscript_data/deseq/data/mouse_only/gavage/volcano/UCPBS_gavvol.tiff", width = 7, height =8, dpi=300)


```

```{r}

#volcano plot - VLP gavage period: HP vs. PBS control comparison

keyvals_HP <- ifelse(
    resultsHPNP_gav$log2FoldChange < -1 & resultsHPNP_gav$padj <0.001, 'black',
      ifelse(resultsHPNP_gav$log2FoldChange > 1 & resultsHPNP_gav$padj <0.001, 'blue',
        'light grey'))
  keyvals_HP[is.na(keyvals_HP)] <- 'light grey'
  names(keyvals_HP)[keyvals_HP == 'blue'] <- 'Over-Abundant HP VLP Scaffolds'
  names(keyvals_HP)[keyvals_HP == 'light grey'] <- 'Non-significant'
  names(keyvals_HP)[keyvals_HP == 'black'] <- 'Under-Abundant HP VLP Scaffolds'

EV_gav_HPNP <- EnhancedVolcano(resultsHPNP_gav, lab=NA , x = 'log2FoldChange',
    y = 'pvalue', pCutoff = 0.001, selectLab = rownames(resultsHPNP_gav)[which(names(keyvals_HP) %in% c('Over-Abundant HP VLP Scaffolds', 'Under-Abundant HP VLP Scaffolds'))], pointSize = 2, gridlines.major = FALSE, gridlines.minor = FALSE, colCustom = keyvals_HP,  legendPosition = 'right', legendLabSize = 12,legendIconSize = 4.0, title = "UCbac + HP VLPs vs. UCbac + PBS",     subtitle = "", caption = ""
)

EV_gav_HPNP


ggsave("/Users/anshul.sinha/Desktop/Sequencing_analyses/phage_metagenomics_rainin_fall_2018/manuscript_data/deseq/data/mouse_only/gavage/volcano/HPPBS_gavvol.tiff", width = 7, height =8, dpi=300)



```




